diff --git a/graal-js/mx.graal-js/suite.py b/graal-js/mx.graal-js/suite.py
index 3093f456ff..ccf8b7740f 100644
--- a/graaljs/graal-js/mx.graal-js/suite.py
+++ b/graaljs/graal-js/mx.graal-js/suite.py
@@ -3,8 +3,8 @@ suite = {
 
   "name" : "graal-js",
 
-  "version" : "22.3.4",
-  "release" : False,
+  "version" : "22.3.5",
+  "release" : True,
   "groupId" : "org.graalvm.js",
   "url" : "http://www.graalvm.org/",
   "developer" : {
