diff --git a/mx.graalpython/suite.py b/mx.graalpython/suite.py
index 79473c646c..3fb4a30ab3 100644
--- a/graalpython/mx.graalpython/suite.py
+++ b/graalpython/mx.graalpython/suite.py
@@ -9,8 +9,8 @@ suite = {
     "name": "graalpython",
     "versionConflictResolution": "latest",
 
-    "version": "22.3.4",
-    "release": False,
+    "version": "22.3.5",
+    "release": True,
     "groupId": "org.graalvm.graalpython",
     "url": "http://www.graalvm.org/",
 
