--- a/graaljs/graal-nodejs/configure
+++ b/graaljs/graal-nodejs/configure
@@ -22,7 +22,7 @@
   from distutils.spawn import find_executable as which

 print('Node.js configure: Found Python {}.{}.{}...'.format(*sys.version_info))
-acceptable_pythons = ((3, 10), (3, 9), (3, 8), (3, 7), (3, 6))
+acceptable_pythons = ((3, 11), (3, 10), (3, 9), (3, 8), (3, 7), (3, 6))
 if sys.version_info[:2] in acceptable_pythons:
   import configure
 else:
