From f61ee5b9ae374188e380e17e84e38010afac9e14 Mon Sep 17 00:00:00 2001
From: Denis Kononenko <denis.kononenko@bell-sw.com>
Date: Wed, 13 Oct 2021 23:20:55 +0000
Subject: [PATCH] fastR: build suite: reduce number of jobs.

---
 mx.fastr/suite.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mx.fastr/suite.py b/mx.fastr/suite.py
index 9b7a6eb..f60f83d 100644
--- a/fastr/mx.fastr/suite.py
+++ b/fastr/mx.fastr/suite.py
@@ -347,7 +347,7 @@ suite = {
         "com.oracle.truffle.r.ffi.impl",
         "com.oracle.truffle.r.launcher"
       ],
-      "max_jobs" : "8",
+      "max_jobs" : "3",
       "native" : True,
       "vpath": True,
       "workingSets" : "FastR",
-- 
2.30.2

