From 79ebdbd1cc5da8245b1599c1885ec982c1484ac5 Mon Sep 17 00:00:00 2001
From: Denis Kononenko <denis.kononenko@bell-sw.com>
Date: Mon, 27 Mar 2023 13:16:40 +0000
Subject: [PATCH] update llvm toolchain

---
 sdk/mx.sdk/suite.py | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/sdk/mx.sdk/suite.py b/sdk/mx.sdk/suite.py
index fd94b76b015..89af463218f 100644
--- a/graal/sdk/mx.sdk/suite.py
+++ b/graal/sdk/mx.sdk/suite.py
@@ -184,11 +184,11 @@ suite = {
       "os_arch" : {
         "linux" : {
           "amd64" : {
-            "urls" : ["{host}/llvm-llvmorg-{version}-linux-amd64.tar.gz"],
-            "sha1" : "4cd45eff8e914189dd8bebcfaf46271c412c57fa",
+            "urls" : ["https://download.bell-sw.com/graalvm/llvm-llvmorg-{version}-x86_64-alpaquita-linux-gnu.tar.gz"],
+            "sha1" : "7b8b1de86d5c9b2b3694fc7b7b36e44389da8ec6",
           },
           "aarch64" : {
-            "urls" : ["{host}/llvm-llvmorg-{version}-linux-aarch64.tar.gz"],
+            "urls" : ["https://download.bell-sw.com/graalvm/llvm-llvmorg-{version}-aarch64-alpaquita-linux-gnu.tar.gz"],
             "sha1" : "bf95d0cb96d29d061e2106f221f9535d38d37daf",
           },
           "riscv64": {
@@ -196,6 +196,16 @@ suite = {
             "sha1" : "4a75da563e277f5d222778f2b814b8e5f7e82609",
           },
         },
+        "linux-musl" : {
+          "amd64" : {
+            "urls" : ["https://download.bell-sw.com/graalvm/llvm-llvmorg-{version}-x86_64-alpaquita-linux-musl.tar.gz"],
+            "sha1" : "e03fde377862aee95a561a044b787e122602d71c",
+          },
+          "aarch64" : {
+            "urls" : ["https://download.bell-sw.com/graalvm/llvm-llvmorg-{version}-aarch64-alpaquita-linux-musl.tar.gz"],
+            "sha1" : "4f73d0cf44cd65b2eb0eb40179503a48825ab043",
+          }
+        },
         "darwin" : {
           "amd64" : {
             "urls" : ["{host}/llvm-llvmorg-{version}-darwin-amd64.tar.gz"],
@@ -232,8 +242,8 @@ suite = {
       "version" : "14.0.6-3-gc7a4a53c32-bgc5e298fd27",
       "host" : "https://lafo.ssw.uni-linz.ac.at/pub/llvm-org",
       "packedResource" : True,
-      "urls" : ["{host}/llvm-src-llvmorg-{version}.tar.gz"],
-      "sha1" : "4b631ecd732e38d491ff6f41da796e393cb1d874",
+      "urls" : ["https://download.bell-sw.com/graalvm/llvm-src-llvmorg-{version}.tar.gz"],
+      "sha1" : "ef41b21c421f6e2a581d702846e2b42aa4057ffc",
       "license" : "Apache-2.0-LLVM",
     },
   },
-- 
2.38.1

