From add996f129f27eb9954427aa79099d8e8c934e81 Mon Sep 17 00:00:00 2001
From: Laurent Bercot <ska-skaware@skarnet.org>
Date: Sat, 10 May 2025 08:05:57 +0000
Subject: [PATCH]  Fix duplicate definition

Signed-off-by: Laurent Bercot <ska@appnovation.com>
---
 src/include/utmps/utmpx.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/include/utmps/utmpx.h b/src/include/utmps/utmpx.h
index fdc1dbc..6e8c240 100644
--- a/src/include/utmps/utmpx.h
+++ b/src/include/utmps/utmpx.h
@@ -82,7 +82,7 @@ extern void logwtmp (char const *, char const *, char const *) ;
 #ifdef _DEFAULT_SOURCE
 extern int utmps_getutxent_r (struct utmpx *, struct utmpx **) ;
 extern int utmps_getutxid_r (struct utmpx const *, struct utmpx *, struct utmpx **) ;
-extern int utmps_getutxent_r (struct utmpx const *, struct utmpx *, struct utmpx **) ;
+extern int utmps_getutxline_r (struct utmpx const *, struct utmpx *, struct utmpx **) ;
 #endif
 
 #ifdef __cplusplus
-- 
2.49.0

