From 2266f1d28e72a5dd1573ce2a5c06d8c691bd817a Mon Sep 17 00:00:00 2001
From: Vasily Isaenko <vasily.isaenko@bell-sw.com>
Date: Sun, 4 Dec 2022 18:43:09 +0000
Subject: [PATCH] async-profiler: corrected location for installed package
 contents

---
 profiler.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/profiler.sh b/profiler.sh
index 73b8790..f910ebc 100755
--- a/profiler.sh
+++ b/profiler.sh
@@ -132,11 +132,11 @@ while [ -h "$SCRIPT_BIN" ]; do
 done
 SCRIPT_DIR="$(cd "$(dirname "$SCRIPT_BIN")" > /dev/null 2>&1; pwd -P)"
 
-JATTACH=$SCRIPT_DIR/build/jattach
-FDTRANSFER=$SCRIPT_DIR/build/fdtransfer
+JATTACH=/opt/async-profiler/jattach
+FDTRANSFER=/opt/async-profiler/fdtransfer
 USE_FDTRANSFER="false"
 FDTRANSFER_PATH=""
-PROFILER=$SCRIPT_DIR/build/libasyncProfiler.so
+PROFILER=/opt/async-profiler/libasyncProfiler.so
 ACTION="collect"
 DURATION="60"
 FILE=""
-- 
2.36.2

