diff --git a/azurelinuxagent/common/osutil/alpine.py b/azurelinuxagent/common/osutil/alpine.py
index a9c73c96..62410179 100644
--- a/azurelinuxagent/common/osutil/alpine.py
+++ b/azurelinuxagent/common/osutil/alpine.py
@@ -32,11 +32,12 @@ class AlpineOSUtil(DefaultOSUtil):
         return True
 
     def get_dhcp_pid(self):
-        return self._get_dhcp_pid(["pidof", "dhcpcd"])
+        return sorted(self._get_dhcp_pid(["pidof", "dhcpcd"]))
 
+    # TODO: We really should get the pid from `dhcpcd --printpidfile`
     def restart_if(self, ifname, retries=None, wait=None):
         logger.info('restarting {} (sort of, actually SIGHUPing dhcpcd)'.format(ifname))
-        pid = self.get_dhcp_pid()
+        pid = self.get_dhcp_pid()[0]
         if pid != None:
             ret = shellutil.run_get_output('kill -HUP {}'.format(pid))  # pylint: disable=W0612
 
diff --git a/azurelinuxagent/common/osutil/factory.py b/azurelinuxagent/common/osutil/factory.py
index fd66fbb0..266941dd 100644
--- a/azurelinuxagent/common/osutil/factory.py
+++ b/azurelinuxagent/common/osutil/factory.py
@@ -79,7 +79,7 @@ def _get_osutil(distro_name, distro_code_name, distro_version, distro_full_name)
 
         return UbuntuOSUtil()
 
-    if distro_name == "alpine":
+    if distro_name in ("alpine", "alpaquita"):
         return AlpineOSUtil()
 
     if distro_name == "kali":
diff --git a/azurelinuxagent/common/version.py b/azurelinuxagent/common/version.py
index 6671a713..36e2c604 100644
--- a/azurelinuxagent/common/version.py
+++ b/azurelinuxagent/common/version.py
@@ -151,6 +151,9 @@ def get_distro():
     if os.path.exists("/etc/photon-release"):
         osinfo[0] = "photonos"
 
+    if os.path.exists('/etc/alpaquita-release'):
+        osinfo[0] = 'alpaquita'
+
     # Remove trailing whitespace and quote in distro name
     osinfo[0] = osinfo[0].strip('"').strip(' ').lower()
     return osinfo
diff --git a/config/alpine/waagent.conf b/config/alpine/waagent.conf
index a8620b5c..87210122 100644
--- a/config/alpine/waagent.conf
+++ b/config/alpine/waagent.conf
@@ -1,5 +1,5 @@
 #
-# Windows Azure Linux Agent Configuration
+# Microsoft Azure Linux Agent Configuration
 #
 
 # Enable extension handling. Do not disable this unless you do not need password reset,
@@ -14,11 +14,11 @@ Provisioning.Agent=auto
 Provisioning.DeleteRootPassword=y
 
 # Generate fresh host key pair.
-Provisioning.RegenerateSshHostKeyPair=y
+Provisioning.RegenerateSshHostKeyPair=n
 
 # Supported values are "rsa", "dsa", "ecdsa", "ed25519", and "auto".
 # The "auto" option is supported on OpenSSH 5.9 (2011) and later.
-Provisioning.SshHostKeyPairType=rsa
+Provisioning.SshHostKeyPairType=auto
 
 # Monitor host name changes and publish changes via DHCP requests.
 Provisioning.MonitorHostName=n
@@ -48,7 +48,7 @@ ResourceDisk.SwapSizeMB=0
 # Comma-separated list of mount options. See mount(8) for valid options.
 ResourceDisk.MountOptions=None
 
-# Respond to load balancer probes if requested by Windows Azure.
+# Respond to load balancer probes if requested by Microsoft Azure.
 LBProbeResponder=y
 
 # Enable logging to serial console (y|n)
@@ -74,6 +74,10 @@ OS.OpensslPath=None
 # Set the path to SSH keys and configuration files
 OS.SshDir=/etc/ssh
 
+# If set, agent will use proxy server to access internet
+#HttpProxy.Host=None
+#HttpProxy.Port=None
+
 # Enable or disable goal state processing auto-update, default is enabled
 # When turned off, it remains on latest version installed on the vm
 # Added this new option AutoUpdate.UpdateToLatestVersion in place of AutoUpdate.Enabled, and encourage users to transition to this new option
@@ -95,4 +99,3 @@ OS.SshDir=/etc/ssh
 
 # Add firewall rules to protect access to Azure host node services
 OS.EnableFirewall=y
-
diff --git a/init/openrc/waagent b/init/openrc/waagent
new file mode 100755
index 00000000..19afb96d
--- /dev/null
+++ b/init/openrc/waagent
@@ -0,0 +1,11 @@
+#!/usr/sbin/openrc-run
+
+name="Microsoft Azure Linux Agent"
+command="/usr/sbin/waagent"
+command_args="-verbose -start"
+pidfile="/var/run/waagent.pid"
+
+depend() {
+    after sshd
+    provide waagent
+}
diff --git a/setup.py b/setup.py
index e83f5989..1e72823c 100755
--- a/setup.py
+++ b/setup.py
@@ -61,6 +61,10 @@ def set_sysv_files(data_files, dest="/etc/rc.d/init.d", src=None):
         src = ["init/waagent"]
     data_files.append((dest, src))
 
+def set_openrc_files(data_files, dest="/etc/init.d", src=None):
+    if src is None:
+        src = ["init/openrc/waagent"]
+    data_files.append((dest, src))
 
 def set_systemd_files(data_files, dest, src=None):
     if src is None:
@@ -254,6 +258,13 @@ def get_data_files(name, version, fullname):  # pylint: disable=R0912
         set_logrotate_files(data_files)
         set_udev_files(data_files)
         set_systemd_files(data_files, dest=systemd_dir_path)
+    elif name in ('alpine', 'alpaquita'):
+        set_bin_files(data_files, dest=agent_bin_path,
+                      src=['bin/waagent'])
+        set_conf_files(data_files, src=["config/alpine/waagent.conf"])
+        set_logrotate_files(data_files)
+        set_udev_files(data_files)
+        set_openrc_files(data_files)
     else:
         # Use default setting
         set_bin_files(data_files, dest=agent_bin_path)
