diff --git a/graal/sdk/mx.sdk/mx_sdk_vm_impl.py b/graal/sdk/mx.sdk/mx_sdk_vm_impl.py
index 88d530ac414..dda7343354f 100644
--- a/graal/sdk/mx.sdk/mx_sdk_vm_impl.py
+++ b/graal/sdk/mx.sdk/mx_sdk_vm_impl.py
@@ -1327,6 +1327,8 @@ class NativePropertiesBuildTask(mx.ProjectBuildTask):
             ] + svm_experimental_options([
                 '-H:+AssertInitializationSpecifiedForAllClasses',
                 '-H:+EnforceMaxRuntimeCompileMethods',
+                '-H:-DeadlockWatchdogExitOnTimeout',
+                '-H:DeadlockWatchdogInterval=0'
             ])
             if _debug_images():
                 build_args += ['-ea', '-O0',] + svm_experimental_options(['-H:+PreserveFramePointer', '-H:-DeleteLocalSymbols'])
