diff --git a/graalpython/mx.graalpython/mx_graalpython.py b/graalpython/mx.graalpython/mx_graalpython.py
index 4e6eedea47..538fa5c8d1 100644
--- a/graalpython/mx.graalpython/mx_graalpython.py
+++ b/graalpython/mx.graalpython/mx_graalpython.py
@@ -2212,6 +2212,8 @@ mx_sdk.register_graalvm_component(mx_sdk.GraalVmLanguage(
                 '-H:+DetectUserDirectoriesInImageHeap',
                 '-H:-CopyLanguageResources',
                 '-Dpolyglot.python.PosixModuleBackend=native',
+                # 9g seems enough, but increased heap size provides safety margin and shorter build times
+                '-J-Xmx11g',
             ],
             language='python',
             default_vm_args=[
