diff --git a/sdk/mx.sdk/suite.py b/sdk/mx.sdk/suite.py
index ca0cedc8bcd..a31e833fbea 100644
--- a/graal/sdk/mx.sdk/suite.py
+++ b/graal/sdk/mx.sdk/suite.py
@@ -163,8 +163,8 @@ suite = {
       "os_arch" : {
         "linux" : {
           "amd64" : {
-            "urls" : ["{host}/llvm-llvmorg-{version}-linux-amd64.tar.gz"],
-            "digest" : "sha512:fdef7aff621c533d30b89c15a3152dd2f6791c99317bb25295e4c9fc9e9b76a341174b55510c6a7e2df7e51be6aad473560927ee5e48605afa884a412147eb8c",
+            "urls" : ["https://packages.bell-sw.com/alpaquita/distfiles/llvm-llvmorg-{version}-alpaquita-glibc-x86_64.tar.xz"],
+            "digest" : "sha512:de88c2d449327a0b12958cf637fd7dca544bf7746027ebb58a6e2f82162af06d533dbc9c3fa8226bca8ab61814be28e1e6cc1ebc7610276b6d303466e93ebfb5",
           },
           "aarch64" : {
             "urls" : ["{host}/llvm-llvmorg-{version}-linux-aarch64.tar.gz"],
@@ -175,6 +175,12 @@ suite = {
             "digest" : "sha512:9186a20d4b657f8a4c86c6730d713c6f8f223a8e9ecceb88d8b5cd3c072e8e0159b810663e57076c0ddcdcd57a819b35b42b543e6633f012175b5f78a6d8de92",
           },
         },
+        "linux-musl" : {
+          "amd64" : {
+            "urls" : ["https://packages.bell-sw.com/alpaquita/distfiles/llvm-llvmorg-{version}-alpaquita-musl-x86_64.tar.xz"],
+            "digest" : "sha512:24951bc469617664a7e2b1bc9e54aa6d10aba8f899d1295f1ab307f0730d69bd43ddc50d608d45f143ded143011dc9d6173aae170563bbac0cb931ab2ad903be",
+          }
+        },
         "darwin" : {
           "amd64" : {
             "urls" : ["{host}/llvm-llvmorg-{version}-darwin-amd64.tar.gz"],
@@ -209,10 +219,10 @@ suite = {
     },
     "LLVM_ORG_SRC" : {
       "version" : "16.0.1-4-gad8c248269-bg39f02d0d6a",
-      "host" : "https://lafo.ssw.uni-linz.ac.at/pub/llvm-org",
+      "host" : "https://download.bell-sw.com/graalvm/llvm",
       "packedResource" : True,
-      "urls" : ["{host}/llvm-src-llvmorg-{version}.tar.gz"],
-      "digest" : "sha512:1bb2f66cc123bb9f0263cd186a8ab7948939f181001e57a7171466534bc89c0ebb17863e90c487f48083f202745ea3d90275a3fa26d793fd2b9f1b62d7e1eabd",
+      "urls" : ["{host}/{version}/llvm-src-llvmorg-{version}.tar.gz"],
+      "digest" : "sha512:aea98b8e8ea32aefc986d3bf50cc71ac9ca13acda3124ed008f8927ece7b9c96f6503baf8bb9ce8a59e82d3ecce6bc874fc1dfbecfe3c73ec3914b2aceda400a",
       "license" : "Apache-2.0-LLVM",
     },
   },
