diff --git a/truffleruby/src/main/c/yarp/Makefile b/truffleruby/src/main/c/yarp/Makefile
index 9639ba8bd20..7fb8c3374a5 100644
--- a/truffleruby/src/main/c/yarp/Makefile
+++ b/truffleruby/src/main/c/yarp/Makefile
@@ -11,7 +11,7 @@ FUZZ_OUTPUT_DIR = $(shell pwd)/fuzz/output
 SOEXT := $(shell ruby -e 'puts RbConfig::CONFIG["SOEXT"]')
 
 CPPFLAGS := -Iinclude
-CFLAGS := -g -O2 -std=c99 -Wall -Werror -Wextra -Wpedantic -Wundef -Wconversion -Wno-missing-braces -fPIC -fvisibility=hidden
+CFLAGS := -g -O2 -std=c99 -Wall -Werror -Wextra -Wundef -Wconversion -Wno-missing-braces -fPIC -fvisibility=hidden
 CC := cc
 WASI_SDK_PATH := /opt/wasi-sdk
 
