diff --git a/ext/standard/tests/file/bug52820.phpt b/ext/standard/tests/file/bug52820.phpt
index 78cfeb61..6ffc3fa9 100644
--- a/ext/standard/tests/file/bug52820.phpt
+++ b/ext/standard/tests/file/bug52820.phpt
@@ -21,6 +21,7 @@ function do_stuff($url) {
     curl_setopt($handle, CURLOPT_VERBOSE, true);
     curl_setopt($handle, CURLOPT_RETURNTRANSFER, true);
     curl_setopt($handle, CURLOPT_STDERR, $o = fopen($url, "w+"));
+    curl_setopt($handle, CURLOPT_NOPROXY, "*");
     curl_exec($handle);
     echo "About to rewind!\n";
     rewind($o);
