From bf9aaf8ef54d325de3843dd004b5e1e66692b0ee Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Fri, 9 Sep 2022 10:03:40 +0300
Subject: [PATCH] don't build static scdoc

---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index 60a80b8..49021fa 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,6 @@
 VERSION=1.11.2
 CFLAGS?=-g
 MAINFLAGS:=-DVERSION='"$(VERSION)"' -Wall -Wextra -Werror -Wno-unused-parameter
-LDFLAGS+=-static
 INCLUDE+=-Iinclude
 PREFIX?=/usr/local
 BINDIR?=$(PREFIX)/bin
-- 
2.25.1

