From 1ffc05767acf99aa4af383aebf3f9fd21ed52648 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Tue, 13 Sep 2022 09:04:30 +0300
Subject: [PATCH] fix ignored return value gcc warning

---
 test/main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/test/main.c b/test/main.c
index 68175df..fc3c98e 100644
--- a/test/main.c
+++ b/test/main.c
@@ -23,6 +23,7 @@
 #include <unistd.h>
 #include <math.h>
 #include <time.h>
+#include <errno.h>
 
 #define pointer_offset(ptr, ofs) (void*)((char*)(ptr) + (ptrdiff_t)(ofs))
 #define pointer_diff(first, second) (ptrdiff_t)((const char*)(first) - (const char*)(second))
@@ -1141,7 +1142,8 @@ test_named_pages(void) {
 	snprintf(name, sizeof(name), "/proc/%d/maps", pid);
 	int fd = open(name, O_RDONLY);
 	if (fd != -1) {
-		read(fd, buf, sizeof(buf));
+		if (read(fd, buf, sizeof(buf)) < 0)
+			printf("read() %s failed: %s", name, strerror(errno));
 		close(fd);
 	}
 #endif
-- 
2.25.1

