diff --git a/lib/tiny-cloud/common b/lib/tiny-cloud/common
index dcc72aa..baa3077 100644
--- a/lib/tiny-cloud/common
+++ b/lib/tiny-cloud/common
@@ -4,7 +4,7 @@
 # set defaults
 [ -f /etc/conf.d/tiny-cloud ] && source /etc/conf.d/tiny-cloud
 : "${CLOUD:=unknown}"
-: "${CLOUD_USER:=alpine}"
+: "${CLOUD_USER:=alpaquita}"
 : "${TINY_CLOUD_LOGS:=/var/log}"
 : "${TINY_CLOUD_VAR:=/var/lib/cloud}"
 
diff --git a/lib/tiny-cloud/tiny-cloud.conf b/lib/tiny-cloud/tiny-cloud.conf
index 5daf7ab..96fc764 100644
--- a/lib/tiny-cloud/tiny-cloud.conf
+++ b/lib/tiny-cloud/tiny-cloud.conf
@@ -6,7 +6,7 @@
 #CLOUD=
 
 # User account where instance SSH keys are installed
-#CLOUD_USER=alpine
+#CLOUD_USER=alpaquita
 
 # IMDS token validity, in seconds (AWS only)
 #IMDS_TOKEN_TTL=5
