From 31d30c3d35012ebe0ee71f410a038676df761aef Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 29 Dec 2021 14:44:55 +0300
Subject: [PATCH] dracut.sh: skip ldconfig on musl systems

---
 dracut.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut.sh b/dracut.sh
index 60ac46f4..ece152c5 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -2226,7 +2226,7 @@ if [[ $kernel_only != yes ]]; then
     fi
 fi
 
-if [[ $kernel_only != yes ]]; then
+if [[ $kernel_only != yes ]] && [[ ! -f /lib/ld-musl-${DRACUT_ARCH:-$(uname -m)}.so.1 ]]; then
     # make sure that library links are correct and up to date
     for f in "$dracutsysrootdir"/etc/ld.so.conf "$dracutsysrootdir"/etc/ld.so.conf.d/*; do
         [[ -f $f ]] && inst_simple "${f#$dracutsysrootdir}"
-- 
2.25.1

