From 149526debf958dfb7a3522749b1aa12249ae60cd Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 29 Dec 2021 14:44:55 +0300
Subject: [PATCH] dracut.sh: skip ldconfig on musl systems

---
 dracut.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut.sh b/dracut.sh
index bbb34697..bc919826 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -2375,7 +2375,7 @@ if [[ $kernel_only != yes ]]; then
     fi
 fi
 
-if [[ $kernel_only != yes ]]; then
+if [[ $kernel_only != yes ]] && [[ ! -f /lib/ld-musl-${DRACUT_ARCH:-$(uname -m)}.so.1 ]]; then
     # make sure that library links are correct and up to date
     build_ld_cache
 fi
-- 
2.25.1

