From ae55a0a527cb3330c1acd8e70228f2763a7f1fd3 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Mon, 10 Jan 2022 12:23:27 +0300
Subject: [PATCH] read busybox/ln help from stderr

---
 dracut-init.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut-init.sh b/dracut-init.sh
index 35c1a48d..353b28cc 100755
--- a/dracut-init.sh
+++ b/dracut-init.sh
@@ -1124,7 +1124,7 @@ instmods() {
     return $_ret
 }
 
-if [[ "$(ln --help)" == *--relative* ]]; then
+if [[ "$(ln --help 2>&1)" == *--relative* ]]; then
     ln_r() {
         local dstdir="${dstdir:-"$initdir"}"
         ln -sfnr "${dstdir}/$1" "${dstdir}/$2"
-- 
2.25.1

