From dec64b275851488c129b6ec3035204745d070bec Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Fri, 25 Jul 2025 12:07:31 +0000
Subject: [PATCH] do not depend on network-legacy (no support)

---
 modules.d/40network/module-setup.sh | 1 -
 1 file changed, 1 deletion(-)

diff --git a/modules.d/40network/module-setup.sh b/modules.d/40network/module-setup.sh
index 101dfb3c..330ea688 100755
--- a/modules.d/40network/module-setup.sh
+++ b/modules.d/40network/module-setup.sh
@@ -23,6 +23,5 @@ depends() {
         fi
     done
 
-    echo "network-legacy"
     return 0
 }
-- 
2.25.1

