From a12905c55ea31f33124dd83153dd36d0e490c500 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Fri, 25 Jul 2025 12:07:31 +0000
Subject: [PATCH] do not depend on network-legacy (no support)

---
 modules.d/40network/module-setup.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/40network/module-setup.sh b/modules.d/40network/module-setup.sh
index 101dfb3c..921c3696 100755
--- a/modules.d/40network/module-setup.sh
+++ b/modules.d/40network/module-setup.sh
@@ -23,6 +23,6 @@ depends() {
         fi
     done
 
-    echo "network-legacy"
+    echo "net-lib kernel-network-modules"
     return 0
 }
-- 
2.25.1

