From 8a0f5fe134e7eef474214b97f2fea4e2fd800d07 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Fri, 24 Oct 2025 08:28:50 +0000
Subject: [PATCH] make using cp in nfs module compatible with busybox

---
 modules.d/95nfs/module-setup.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules.d/95nfs/module-setup.sh b/modules.d/95nfs/module-setup.sh
index b34c75c0..9478b279 100755
--- a/modules.d/95nfs/module-setup.sh
+++ b/modules.d/95nfs/module-setup.sh
@@ -121,13 +121,13 @@ install() {
 
     # use the same directory permissions as the host
     [[ -d "$dracutsysrootdir"/var/lib/nfs/statd ]] \
-        && $DRACUT_CP -L --preserve=ownership -t "$initdir"/var/lib/nfs "$dracutsysrootdir"/var/lib/nfs/statd \
+        && $DRACUT_CP -L -p -t "$initdir"/var/lib/nfs "$dracutsysrootdir"/var/lib/nfs/statd \
         && rm -rf "$initdir"/var/lib/nfs/statd/*
     [[ -d "$dracutsysrootdir"/var/lib/nfs/statd/sm ]] \
-        && $DRACUT_CP -L --preserve=ownership -t "$initdir"/var/lib/nfs/statd "$dracutsysrootdir"/var/lib/nfs/statd/sm \
+        && $DRACUT_CP -L -p -t "$initdir"/var/lib/nfs/statd "$dracutsysrootdir"/var/lib/nfs/statd/sm \
         && rm -rf "$initdir"/var/lib/nfs/statd/sm/*
     [[ -d "$dracutsysrootdir"/var/lib/nfs/sm ]] \
-        && $DRACUT_CP -L --preserve=ownership -t "$initdir"/var/lib/nfs "$dracutsysrootdir"/var/lib/nfs/sm \
+        && $DRACUT_CP -L -p -t "$initdir"/var/lib/nfs "$dracutsysrootdir"/var/lib/nfs/sm \
         && rm -rf "$initdir"/var/lib/nfs/sm/*
 
     # Rather than copy the passwd file in, just set a user for rpcbind
-- 
2.25.1

