From 62162a017fcaa80a487ebc4828288acd3c30f8a7 Mon Sep 17 00:00:00 2001
From: Vasily Isaenko <vasily.isaenko@bell-sw.com>
Date: Thu, 5 May 2022 14:00:40 +0000
Subject: [PATCH] Added Alpaquita distribution

---
 alsaconf/alsaconf.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/alsaconf/alsaconf.in b/alsaconf/alsaconf.in
index 3d6adca..af11244 100644
--- a/alsaconf/alsaconf.in
+++ b/alsaconf/alsaconf.in
@@ -65,6 +65,8 @@ elif test -f /etc/redhat-release && grep -q "Red Hat" /etc/redhat-release; then
   distribution="redhat"
 elif test -f /etc/fedora-release && grep -q "Fedora" /etc/fedora-release; then
   distribution="fedora"
+elif test -f /etc/alpaquita-release; then
+  distribution="alpaquita"
 elif [ -f /etc/slackware-version -o -f /etc/slamd64-version ]; then
   distribution="slackware"
 else
-- 
2.34.1

