diff --git a/alsaconf/alsaconf.in b/alsaconf/alsaconf.in
index 3d6adca..af11244 100644
--- a/alsaconf/alsaconf.in
+++ b/alsaconf/alsaconf.in
@@ -67,6 +67,8 @@
   distribution="fedora"
 elif test -f /etc/alpine-release; then
   distribution="alpine"
+elif test -f /etc/alpaquita-release; then
+  distribution="alpaquita"
 elif [ -f /etc/slackware-version -o -f /etc/slamd64-version ]; then
   distribution="slackware"
 else
-- 
2.34.1

