Failing test: long-dns-crq

diff --git a/tests/cert-tests/Makefile.am b/tests/cert-tests/Makefile.am
index c85361e..34a3585 100644
--- a/tests/cert-tests/Makefile.am
+++ b/tests/cert-tests/Makefile.am
@@ -116,7 +116,7 @@ dist_check_SCRIPTS = pathlen.sh aki.sh invalid-sig.sh email.sh \
 	pkcs8-eddsa.sh certtool-subca.sh certtool-verify-profiles.sh x509-duplicate-ext.sh x25519-and-x448.sh
 
 dist_check_SCRIPTS += key-id.sh ecdsa.sh pkcs8-invalid.sh key-invalid.sh pkcs8-decode.sh pkcs8.sh pkcs8-eddsa.sh \
-	certtool-utf8.sh crq.sh
+	certtool-utf8.sh
 
 if STRICT_DER_TIME
 dist_check_SCRIPTS += cert-non-digits-time.sh reject-invalid-time.sh
