diff --git a/tests/Makefile.in b/tests/Makefile.in
index 716b17a..28b4ed6 100644
--- a/tests/Makefile.in
+++ b/tests/Makefile.in
@@ -1480,7 +1480,7 @@ mini_dtls_fork_DEPENDENCIES = $(COMMON_GNUTLS_LDADD) libutils.la \
 mini_dtls_fragments_SOURCES = mini-dtls-fragments.c
 mini_dtls_fragments_OBJECTS =  \
 	mini_dtls_fragments-mini-dtls-fragments.$(OBJEXT)
-mini_dtls_fragments_LDADD = $(LDADD)
+mini_dtls_fragments_LDADD = $(LDADD) ../gl/libgnu.la
 mini_dtls_fragments_DEPENDENCIES = $(COMMON_GNUTLS_LDADD) libutils.la \
 	$(am__DEPENDENCIES_2)
 mini_dtls_heartbeat_SOURCES = mini-dtls-heartbeat.c
