diff --git a/make/autoconf/toolchain.m4 b/make/autoconf/toolchain.m4
index af6a080..9c26a7b 100644
--- a/make/autoconf/toolchain.m4
+++ b/make/autoconf/toolchain.m4
@@ -751,7 +751,7 @@ AC_DEFUN_ONCE([TOOLCHAIN_DETECT_TOOLCHAIN_EXTRA],
   if test "x$OPENJDK_TARGET_OS" != xwindows; then
     UTIL_LOOKUP_TOOLCHAIN_PROGS(STRIP, strip)
     if test "x$TOOLCHAIN_TYPE" = xgcc; then
-      UTIL_LOOKUP_TOOLCHAIN_PROGS(NM, nm gcc-nm)
+      UTIL_LOOKUP_TOOLCHAIN_PROGS(NM, gcc-nm nm)
     else
       UTIL_LOOKUP_TOOLCHAIN_PROGS(NM, nm)
     fi
