diff --git a/src/test/signal.sl b/src/test/signal.sl
index b1c05a8..30a067d 100644
--- a/src/test/signal.sl
+++ b/src/test/signal.sl
@@ -18,6 +18,7 @@ private define test_signal (sig, func, old_func)
    failed ("signal ($sig,$func,$old_func), got $old"$);
 }
 
+signal(SIGHUP, SIG_DFL);
 test_signal (SIGHUP, &handle_hup, SIG_DFL);
 test_signal (SIGHUP, SIG_APP, &handle_hup);
 test_signal (SIGHUP, SIG_DFL, SIG_DFL);
