From ef64fc08e5f58a1cbe12635a641dd5443d9a309d Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Tue, 5 Jul 2022 11:20:58 +0300
Subject: [PATCH] load libz to fix test on glibc

---
 test/sysdep/ffi_lib_c.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/sysdep/ffi_lib_c.lua b/test/sysdep/ffi_lib_c.lua
index a368d75..2a01b8a 100644
--- a/test-suite/test/sysdep/ffi_lib_c.lua
+++ b/test-suite/test/sysdep/ffi_lib_c.lua
@@ -81,7 +81,7 @@ end
 
 do
   if not (ffi.os == "Windows" or ffi.os == "Other") then
-    ffi.load("pthread")
+    ffi.load("libz")
   end
 end
 
-- 
2.25.1

