From 3cd1f7fc052b1471c915b7ac75e5bb7910bf7eec Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Mon, 15 May 2023 14:20:26 +0000
Subject: [PATCH] init: re-exec init before shutdown/reboot

---
 init/init.c | 32 +++++++++++++++++++++++++++-----
 1 file changed, 27 insertions(+), 5 deletions(-)

diff --git a/init/init.c b/init/init.c
index 1e1ce833d..71a4afa80 100644
--- a/init/init.c
+++ b/init/init.c
@@ -810,7 +810,7 @@ static void run_shutdown_and_kill_processes(void)
 static void halt_reboot_pwoff(int sig) NORETURN;
 static void halt_reboot_pwoff(int sig)
 {
-	const char *m;
+	const char *m, *cmd;
 	unsigned rb;
 
 	/* We may call run() and it unmasks signals,
@@ -821,18 +821,24 @@ static void halt_reboot_pwoff(int sig)
 	 */
 	reset_sighandlers_and_unblock_sigs();
 
-	run_shutdown_and_kill_processes();
-
 	m = "halt";
 	rb = RB_HALT_SYSTEM;
+	cmd = "init do_halt";
 	if (sig == SIGTERM) {
 		m = "reboot";
 		rb = RB_AUTOBOOT;
+		cmd = "init do_reboot";
 	} else if (sig == SIGUSR2) {
 		m = "poweroff";
 		rb = RB_POWER_OFF;
+		cmd = "init do_poweroff";
 	}
+
 	message(L_CONSOLE, "Requesting system %s", m);
+	init_exec(cmd);
+
+	message(L_LOG | L_CONSOLE, "Error requesting system %s via init re-exec", m);
+	run_shutdown_and_kill_processes();
 	pause_and_low_level_reboot(rb);
 	/* not reached */
 }
@@ -1039,6 +1045,7 @@ int init_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int init_main(int argc UNUSED_PARAM, char **argv)
 {
 	struct sigaction sa;
+	unsigned init_restart_action = 0;
 
 	INIT_G();
 
@@ -1114,8 +1121,16 @@ int init_main(int argc UNUSED_PARAM, char **argv)
 	 */
 	//putenv((char *) "HOME=/");
 
-	if (argv[1])
-		xsetenv("RUNLEVEL", argv[1]);
+	if (argv[1]) {
+		if (strcmp(argv[1], "do_reboot") == 0)
+			init_restart_action = RB_AUTOBOOT;
+		else if (strcmp(argv[1], "do_poweroff") == 0)
+			init_restart_action = RB_POWER_OFF;
+		else if (strcmp(argv[1], "do_halt") == 0)
+			init_restart_action = RB_HALT_SYSTEM;
+		else
+			xsetenv("RUNLEVEL", argv[1]);
+	}
 
 #if !ENABLE_FEATURE_INIT_QUIET
 	/* Hello world */
@@ -1163,6 +1178,13 @@ int init_main(int argc UNUSED_PARAM, char **argv)
 		nuke_str(*argv);
 #endif
 
+	if (init_restart_action) {
+		reset_sighandlers_and_unblock_sigs();
+		run_shutdown_and_kill_processes();
+		pause_and_low_level_reboot(init_restart_action);
+		/* not reached */
+	}
+
 	/* Set up STOP signal handlers */
 	/* Stop handler must allow only SIGCONT inside itself */
 	memset(&sa, 0, sizeof(sa));
-- 
2.25.1

