diff -r 54aa3fd1151f make/linux/makefiles/gcc.make
--- a/hotspot/make/linux/makefiles/gcc.make	Fri Nov 06 03:45:25 2020 +0000
+++ b/hotspot/make/linux/makefiles/gcc.make	Mon Dec 07 13:22:36 2020 +0300
@@ -202,7 +202,7 @@
 endif
 
 # Compiler warnings are treated as errors
-WARNINGS_ARE_ERRORS = -Werror
+# WARNINGS_ARE_ERRORS = -Werror
 
 ifeq ($(USE_CLANG), true)
   # However we need to clean the code up before we can unrestrictedly enable this option with Clang
