From 2cc6154609594ed722465c7ed616a13e2b599cf1 Mon Sep 17 00:00:00 2001
From: Denis Kononenko <denis.kononenko@bell-sw.com>
Date: Sun, 18 Sep 2022 17:43:55 +0000
Subject: [PATCH] Allow build from sources out of the version control system.

---
 .../processor/BuildInformationProcessor.java           | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/processor/java/org/truffleruby/processor/BuildInformationProcessor.java b/src/processor/java/org/truffleruby/processor/BuildInformationProcessor.java
index a3b9405a45..767a746305 100644
--- a/truffleruby/src/processor/java/org/truffleruby/processor/BuildInformationProcessor.java
+++ b/truffleruby/src/processor/java/org/truffleruby/processor/BuildInformationProcessor.java
@@ -56,9 +56,15 @@ public class BuildInformationProcessor extends TruffleRubyProcessor {
         try {
             trufflerubyHome = findHome();
             buildName = System.getenv("TRUFFLERUBY_BUILD_NAME");
-            fullRevision = runCommand("git rev-parse HEAD");
+            fullRevision = System.getenv("TRUFFLERUBY_COMMIT_SHA");
+	     if (fullRevision == null || fullRevision.isEmpty()) {
+                fullRevision = runCommand("git rev-parse HEAD");
+            }
             shortRevision = fullRevision.substring(0, 8);
-            compileDate = runCommand("git log -1 --date=short --pretty=format:%cd");
+            compileDate = System.getenv("TRUFFLERUBY_COMMIT_DATE");
+	     if (compileDate == null || compileDate.isEmpty()) {
+                compileDate = runCommand("git log -1 --date=short --pretty=format:%cd");
+	     }
             copyrightYear = compileDate.split("\\-")[0];
             kernelMajorVersion = findKernelMajorVersion();
         } catch (Throwable e) {
-- 
2.34.4

