diff --git a/mx.mx/suite.py b/mx.mx/suite.py
index d488305..2a02bc0 100644
--- a/mx/mx.mx/suite.py
+++ b/mx/mx.mx/suite.py
@@ -537,6 +537,9 @@ suite = {
           "amd64" : {
             "urls" : ["https://lafo.ssw.jku.at/pub/graal-external-deps/ninja-{version}-linux-amd64-musl.zip"],
             "sha1" : "0cc6d5cff72e63444b7abb3fc0562f6e70089147"
+          },
+          "<others>" : {
+            "optional" : True
           }
         },
         "darwin" : {
