diff --git a/sdk/mx.sdk/mx_sdk_vm_impl.py b/sdk/mx.sdk/mx_sdk_vm_impl.py
index e9e1978c336..c399f7142e8 100644
--- a/graal/sdk/mx.sdk/mx_sdk_vm_impl.py
+++ b/graal/sdk/mx.sdk/mx_sdk_vm_impl.py
@@ -1277,6 +1277,8 @@ class NativePropertiesBuildTask(mx.ProjectBuildTask):
                 '-H:+AssertInitializationSpecifiedForAllClasses',
                 '-H:+EnforceMaxRuntimeCompileMethods',
                 '-Dorg.graalvm.version={}'.format(_suite.release_version()),
+                '-H:-DeadlockWatchdogExitOnTimeout',
+                '-H:DeadlockWatchdogInterval=0'
             ]
             if _debug_images():
                 build_args += ['-ea', '-H:-AOTInline', '-H:+PreserveFramePointer', '-H:-DeleteLocalSymbols']
