From 84daee7fbdd6b1333ec51c6123507884448a094a Mon Sep 17 00:00:00 2001
From: Denis Kononenko <denis.kononenko@bell-sw.com>
Date: Mon, 19 Sep 2022 18:51:27 +0000
Subject: [PATCH] rubyvm unsupported methods

---
 mx.truffleruby/mx_truffleruby.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mx.truffleruby/mx_truffleruby.py b/mx.truffleruby/mx_truffleruby.py
index 8b86eae585..d87b99313b 100644
--- a/truffleruby/mx.truffleruby/mx_truffleruby.py
+++ b/truffleruby/mx.truffleruby/mx_truffleruby.py
@@ -276,7 +276,8 @@ mx_sdk_vm.register_graalvm_component(mx_sdk_vm.GraalVmLanguage(
             build_args=[
                 '-H:+DumpThreadStacksOnSignal',
                 '-H:+DetectUserDirectoriesInImageHeap',
-                '-H:+TruffleCheckBlockListMethods'
+                '-H:+TruffleCheckBlockListMethods',
+                '--report-unsupported-elements-at-runtime'
             ],
             language='ruby',
             option_vars=[
-- 
2.34.4

