From f9efe63c88e72636a1d855d9f4bc7848db4088d6 Mon Sep 17 00:00:00 2001
From: Denis Kononenko <denis.kononenko@bell-sw.com>
Date: Mon, 19 Sep 2022 18:37:41 +0000
Subject: [PATCH] update llvm toolchain

---
 sdk/mx.sdk/suite.py | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/sdk/mx.sdk/suite.py b/sdk/mx.sdk/suite.py
index 0a8db186431..d18b321754e 100644
--- a/graal/sdk/mx.sdk/suite.py
+++ b/graal/sdk/mx.sdk/suite.py
@@ -150,14 +150,24 @@ suite = {
       "os_arch" : {
         "linux" : {
           "amd64" : {
-            "urls" : ["{host}/llvm-llvmorg-{version}-linux-amd64.tar.gz"],
-            "sha1" : "f32dc5497d9dd679f9ef42fd26918c3ad659b8e3",
+            "urls" : ["https://download.bell-sw.com/graalvm/llvm-llvmorg-{version}-x86_64-alpaquita-linux-gnu.tar.gz"],
+            "sha1" : "9e4f6a3b03563c9e38ddc6b8d20b237949f75601",
           },
           "aarch64" : {
-            "urls" : ["{host}/llvm-llvmorg-{version}-linux-aarch64.tar.gz"],
+            "urls" : ["https://download.bell-sw.com/graalvm/llvm-llvmorg-{version}-aarch64-alpaquita-linux-gnu.tar.gz"],
             "sha1" : "8bf8d3f96b6be5d361e809d1928c3b6c46791dfe",
           }
         },
+        "linux-musl" : {
+          "amd64" : {
+            "urls" : ["https://download.bell-sw.com/graalvm/llvm-llvmorg-{version}-x86_64-alpaquita-linux-musl.tar.gz"],
+            "sha1" : "17c98a3c6a41c7043bd197412c7870dff6bb4641",
+          },
+          "aarch64" : {
+            "urls" : ["https://download.bell-sw.com/graalvm/llvm-llvmorg-{version}-aarch64-alpaquita-linux-musl.tar.gz"],
+            "sha1" : "4f73d0cf44cd65b2eb0eb40179503a48825ab043",
+          }
+        },
         "darwin" : {
           "amd64" : {
             "urls" : ["{host}/llvm-llvmorg-{version}-darwin-amd64.tar.gz"],
@@ -194,8 +204,8 @@ suite = {
       "version" : "14.0.3-2-g772a7a659e-bgd661ed0276",
       "host" : "https://lafo.ssw.uni-linz.ac.at/pub/llvm-org",
       "packedResource" : True,
-      "urls" : ["{host}/llvm-src-llvmorg-{version}.tar.gz"],
-      "sha1" : "f762ff022e26af02b0fbb3176d7280bc4809f562",
+      "urls" : ["https://download.bell-sw.com/graalvm/llvm-src-llvmorg-{version}.tar.gz"],
+      "sha1" : "cef69a6f4f01bfdfaa8519c2899342b84f690588",
       "license" : "Apache-2.0-LLVM",
     },
   },
-- 
2.34.4

