From 303b50088a7534c709b2eacf5149fccd39b1caa1 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Fri, 11 Feb 2022 18:47:02 +0300
Subject: [PATCH] init.d/agetty: depend on hostname (to show in prompt)

---
 init.d/agetty.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/init.d/agetty.in b/init.d/agetty.in
index 329cd214..99528387 100644
--- a/init.d/agetty.in
+++ b/init.d/agetty.in
@@ -14,12 +14,12 @@ supervisor=supervise-daemon
 port="${RC_SVCNAME#*.}"
 respawn_period="${respawn_period:-60}"
 term_type="${term_type:-linux}"
-command=/sbin/agetty
+command=/usr/sbin/agetty
 command_args_foreground="${agetty_options} ${port} ${baud} ${term_type}"
 pidfile="/run/${RC_SVCNAME}.pid"
 
 depend() {
-	after local
+	after local hostname
 	keyword -prefix
 	provide getty
 }
-- 
2.25.1

