diff -r 405102e26a62 src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.runtime/src/jdk/vm/ci/runtime/JVMCI.java
--- a/src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.runtime/src/jdk/vm/ci/runtime/JVMCI.java	Wed Dec 23 09:07:55 2020 +0100
+++ b/src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.runtime/src/jdk/vm/ci/runtime/JVMCI.java	Mon Feb 08 11:11:51 2021 +0300
@@ -24,9 +24,11 @@
 
 import java.util.Formatter;
 
+import jdk.vm.ci.common.NativeImageReinitialize;
+
 public class JVMCI {
 
-    private static final JVMCIRuntime runtime;
+    @NativeImageReinitialize private static final JVMCIRuntime runtime;
 
     private static native JVMCIRuntime initializeRuntime();
 
