Fix format since we don't include a revision number

diff --git a/jemalloc.pc.in b/jemalloc.pc.in
index b50770d1..9a58e92d 100644
--- a/jemalloc.pc.in
+++ b/jemalloc.pc.in
@@ -8,7 +8,7 @@ jemalloc_prefix=@JEMALLOC_PREFIX@
 Name: jemalloc
 Description: A general purpose malloc(3) implementation that emphasizes fragmentation avoidance and scalable concurrency support.
 URL: https://jemalloc.net/
-Version: @jemalloc_version_major@.@jemalloc_version_minor@.@jemalloc_version_bugfix@_@jemalloc_version_nrev@
+Version: @jemalloc_version_major@.@jemalloc_version_minor@.@jemalloc_version_bugfix@
 Cflags: -I${includedir}
 Libs: -L${libdir} -ljemalloc${install_suffix}
 Libs.private: @LIBS@
