--- a/ext/sockets/tests/socket_cmsg_credentials.phpt	2022-07-12 18:37:03.208390511 +0300
+++ b/ext/sockets/tests/socket_cmsg_credentials.phpt	2022-07-12 18:38:39.460513712 +0300
@@ -17,7 +17,7 @@
 --FILE--
 <?php
 include __DIR__."/mcast_helpers.php.inc";
-$path = __DIR__ . "/socket_cmsg_credentials.sock";
+$path = __DIR__ . "/sk";
 
 @unlink($path);
 

--- a/ext/sockets/tests/socket_cmsg_rights.phpt	2022-07-12 18:37:03.208390511 +0300
+++ b/ext/sockets/tests/socket_cmsg_rights.phpt	2022-07-12 18:38:27.100498187 +0300
@@ -18,7 +18,7 @@
 --FILE--
 <?php
 include __DIR__."/mcast_helpers.php.inc";
-$path = __DIR__ . "/socket_cmsg_rights.sock";
+$path = __DIR__ . "/sk";
 
 @unlink($path);
 
