--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 LAYOUT ?= linux
 SCDOC := scdoc
 LIBBSD_CFLAGS =
-LIBBSD_LIBS =
+LIBBSD_LIBS = -lbsd
 
 PACKAGE_NAME := ifupdown-ng
 PACKAGE_VERSION := 0.11.3
@@ -17,6 +17,7 @@
 CFLAGS ?= -ggdb3 -Os
 CFLAGS += -Wall -Wextra -Werror
 CFLAGS += -Wmissing-declarations -Wmissing-prototypes -Wcast-align -Wpointer-arith -Wreturn-type
+CFLAGS += -Wno-error=implicit-function-declaration
 CFLAGS += ${LIBBSD_CFLAGS}
 CPPFLAGS = -I.
 CPPFLAGS += -DINTERFACES_FILE=\"${INTERFACES_FILE}\"
