diff --git a/mx.fastr/suite.py b/mx.fastr/suite.py
index 97e4c4957b..549b88844c 100644
--- a/fastr/mx.fastr/suite.py
+++ b/fastr/mx.fastr/suite.py
@@ -411,11 +411,22 @@ suite = {
         "com.oracle.truffle.r.ffi.impl",
         "com.oracle.truffle.r.launcher"
       ],
-      "max_jobs" : "8",
       "native" : True,
       "vpath": True,
       "workingSets" : "FastR",
       "buildDependencies" : ["FASTR"],
+      "os_arch" : {
+        "linux-musl" : {
+          "<others>" : {
+            "max_jobs" : "2",
+          },
+        },
+        "<others>" : {
+          "<others>" : {
+            "max_jobs" : "3",
+          }
+        }
+      }
     },
 
     "com.oracle.truffle.r.test.tck" : {

