diff --git a/sdk/mx.sdk/mx_sdk_vm_impl.py b/sdk/mx.sdk/mx_sdk_vm_impl.py
index ecfbf595451..5d97db2606c 100644
--- a/graal/sdk/mx.sdk/mx_sdk_vm_impl.py
+++ b/graal/sdk/mx.sdk/mx_sdk_vm_impl.py
@@ -3748,9 +3748,8 @@ def graalvm_vendor_version():
     # Examples:
     # GraalVM CE 17.0.1+4.1
     # Oracle GraalVM 17.0.1+4.1
-    return '{vendor} {version}'.format(
-        vendor=('Oracle ' + _graalvm_base_name) if mx_sdk_vm.ee_implementor() else (_graalvm_base_name + ' CE'),
-        version=graalvm_version(version_type='vendor')
+    return '{vendor}'.format(
+        vendor=('Oracle ' + _graalvm_base_name) if mx_sdk_vm.ee_implementor() else ('Liberica-NIK-23.0.5-1')
     )
 
 
