--- a/graal/sdk/mx.sdk/suite.py
+++ b/graal/sdk/mx.sdk/suite.py
@@ -150,18 +150,28 @@
       "os_arch" : {
         "linux" : {
           "amd64" : {
-            "urls" : ["{host}/llvm-llvmorg-{version}-linux-amd64.tar.gz"],
-            "digest" : "sha512:10879717ca1e09766c86f1c11789dc8937b211d2319bec8c3a7f08f71585dca0ddd0e819f020233e498334b5df7908500bec3f8f6c695877b87b9c3a8e071c5d",
+            "urls" : ["https://packages.bell-sw.com/alpaquita/distfiles/llvm-llvmorg-{version}-alpaquita-glibc-x86_64.tar.xz"],
+            "digest" : "sha512:4cade99fbb49e86b0efc0ff2d3dd9bd81b998e2616bf0a0ff48d3a2a42ec980ee37e5e497c9e881c5cec5383b39328795fd7201d528b4a0fe31e7bc2afaf9b9f",
           },
           "aarch64" : {
-            "urls" : ["{host}/llvm-llvmorg-{version}-linux-aarch64.tar.gz"],
-            "digest" : "sha512:6b5453d032c5acc0c70bd34781b8f4251a8a2690ffcd7ab5ac0ad267f3062d4396956411f113e83aba637c4012901ea983d45083ac45d099f0a32046744ba163",
+            "urls" : ["https://packages.bell-sw.com/alpaquita/distfiles/llvm-llvmorg-{version}-alpaquita-linux-gnu-aarch64.tar.xz"],
+            "digest" : "sha512:d2cb16baace7bad3c8c232a8ca5ea4edb007e2506e8771cb14ed39ff7f25f639f808263f648394cdfebad9178e28f0169342298139306f5a566ea439ae3993f6",
           },
           "riscv64": {
             "urls" : ["{host}/llvm-llvmorg-{version}-linux-riscv64.tar.gz"],
             "digest" : "sha512:3bdc5c073a903cb74b4143108fc34b109759fdb715596394e47a92c108cc191b0f32cd52ea49de54e7825cbf7acb6af39b9a1ce1f3a1e00aec144109f1289f64",
           },
         },
+        "linux-musl" : {
+          "amd64" : {
+            "urls" : ["https://packages.bell-sw.com/alpaquita/distfiles/llvm-llvmorg-{version}-alpaquita-musl-x86_64.tar.xz"],
+            "digest" : "sha512:f11063e8196ed6e74c45d9c14f2ad31a7ebbd0e3191b36d100af480a7a91070f11b058134ef7e69ecfbce980655d60edac4add982a7c81f37de1a70ca3346b3c",
+          },
+          "aarch64" : {
+            "urls" : ["https://packages.bell-sw.com/alpaquita/distfiles/llvm-llvmorg-{version}-alpaquita-linux-musl-aarch64.tar.xz"],
+            "digest" : "sha512:eff36f80a0dc68732641e01a508aeafe6550c5e7637150d2df28914bd0935b9a4bb54b680f59b658d6a6d55960bda0a08a98683316f257190a3c198a4a8e9089",
+          },
+        },
         "darwin" : {
           "amd64" : {
             "urls" : ["{host}/llvm-llvmorg-{version}-darwin-amd64.tar.gz"],
@@ -198,8 +208,8 @@
       "version" : "15.0.6-4-gaeae49c2d7-bg8671e93c04",
       "host" : "https://lafo.ssw.uni-linz.ac.at/pub/llvm-org",
       "packedResource" : True,
-      "urls" : ["{host}/llvm-src-llvmorg-{version}.tar.gz"],
-      "digest" : "sha512:0fa2843843ebea5cf8d9dd6de699b2c09bcbf0ecaad3caca64258966b2cb09ea3c03ad06b19683b97a7ce2462d679e83fe69c5f54905371e4011211d458f039f",
+      "urls" : ["https://download.bell-sw.com/graalvm/llvm/{version}/llvm-src-llvmorg-{version}.tar.gz"],
+      "digest" : "sha512:1e16c09738fe4ee5ebdfc85c57e1d1206548310f939c4270d9b908968fb2e6c32724fd60b51736bd90c797375676a55a599fc2cb819eb1523f747f584214ebb7",
       "license" : "Apache-2.0-LLVM",
     },
   },
