diff --git a/compiler/mx.compiler/suite.py b/compiler/mx.compiler/suite.py
index 5372ff7872b..c4d842b5762 100644
--- a/graal/compiler/mx.compiler/suite.py
+++ b/graal/compiler/mx.compiler/suite.py
@@ -5,7 +5,7 @@ suite = {
 
   "groupId" : "org.graalvm.compiler",
   "version" : "23.0.4",
-  "release" : False,
+  "release" : True,
   "url" : "http://www.graalvm.org/",
   "developer" : {
     "name" : "GraalVM Development",
diff --git a/espresso/mx.espresso/suite.py b/espresso/mx.espresso/suite.py
index f1408521b7a..7f864569188 100644
--- a/graal/espresso/mx.espresso/suite.py
+++ b/graal/espresso/mx.espresso/suite.py
@@ -24,7 +24,7 @@ suite = {
     "mxversion": "6.17.0",
     "name": "espresso",
     "version" : "23.0.4",
-    "release" : False,
+    "release" : True,
     "groupId" : "org.graalvm.espresso",
     "url" : "https://www.graalvm.org/reference-manual/java-on-truffle/",
     "developer" : {
diff --git a/regex/mx.regex/suite.py b/regex/mx.regex/suite.py
index 5ddf3fd01b0..0ff089f254b 100644
--- a/graal/regex/mx.regex/suite.py
+++ b/graal/regex/mx.regex/suite.py
@@ -44,7 +44,7 @@ suite = {
   "name" : "regex",
 
   "version" : "23.0.4",
-  "release" : False,
+  "release" : True,
   "groupId" : "org.graalvm.regex",
   "url" : "http://www.graalvm.org/",
   "developer" : {
diff --git a/sdk/mx.sdk/suite.py b/sdk/mx.sdk/suite.py
index 0afe2abc071..923282304bd 100644
--- a/graal/sdk/mx.sdk/suite.py
+++ b/graal/sdk/mx.sdk/suite.py
@@ -42,7 +42,7 @@ suite = {
   "mxversion": "6.17.0",
   "name" : "sdk",
   "version" : "23.0.4",
-  "release" : False,
+  "release" : True,
   "sourceinprojectwhitelist" : [],
   "url" : "https://github.com/oracle/graal",
   "groupId" : "org.graalvm.sdk",
diff --git a/substratevm/mx.substratevm/suite.py b/substratevm/mx.substratevm/suite.py
index e29ed921625..300f84ca71f 100644
--- a/graal/substratevm/mx.substratevm/suite.py
+++ b/graal/substratevm/mx.substratevm/suite.py
@@ -3,7 +3,7 @@ suite = {
     "mxversion": "6.17.0",
     "name": "substratevm",
     "version" : "23.0.4",
-    "release" : False,
+    "release" : True,
     "url" : "https://github.com/oracle/graal/tree/master/substratevm",
 
     "groupId" : "org.graalvm.nativeimage",
diff --git a/tools/mx.tools/suite.py b/tools/mx.tools/suite.py
index 81172a2726f..1e94961438a 100644
--- a/graal/tools/mx.tools/suite.py
+++ b/graal/tools/mx.tools/suite.py
@@ -27,7 +27,7 @@ suite = {
 
     "groupId" : "org.graalvm.tools",
     "version" : "23.0.4",
-    "release" : False,
+    "release" : True,
     "url" : "http://openjdk.java.net/projects/graal",
     "developer" : {
         "name" : "GraalVM Development",
diff --git a/truffle/mx.truffle/suite.py b/truffle/mx.truffle/suite.py
index 1f816e8e86e..a7a21a776cf 100644
--- a/graal/truffle/mx.truffle/suite.py
+++ b/graal/truffle/mx.truffle/suite.py
@@ -42,7 +42,7 @@ suite = {
   "mxversion": "6.17.0",
   "name" : "truffle",
   "version" : "23.0.4",
-  "release" : False,
+  "release" : True,
   "groupId" : "org.graalvm.truffle",
   "sourceinprojectwhitelist" : [],
   "url" : "http://openjdk.java.net/projects/graal",
diff --git a/vm/mx.vm/suite.py b/vm/mx.vm/suite.py
index 8d17caef609..5dca385ad42 100644
--- a/graal/vm/mx.vm/suite.py
+++ b/graal/vm/mx.vm/suite.py
@@ -2,7 +2,7 @@ suite = {
     "name": "vm",
     "version" : "23.0.4",
     "mxversion": "6.17.0",
-    "release" : False,
+    "release" : True,
     "groupId" : "org.graalvm",
 
     "url" : "http://www.graalvm.org/",
