diff --git a/graal/compiler/mx.compiler/suite.py b/graal/compiler/mx.compiler/suite.py
index 5372ff7872b..ee9f4f02ec3 100644
--- a/graal/compiler/mx.compiler/suite.py
+++ b/graal/compiler/mx.compiler/suite.py
@@ -4,8 +4,8 @@ suite = {
   "sourceinprojectwhitelist" : [],
 
   "groupId" : "org.graalvm.compiler",
-  "version" : "23.0.4",
-  "release" : False,
+  "version" : "23.0.5",
+  "release" : True,
   "url" : "http://www.graalvm.org/",
   "developer" : {
     "name" : "GraalVM Development",
diff --git a/graal/espresso/mx.espresso/suite.py b/graal/espresso/mx.espresso/suite.py
index f1408521b7a..3abcec7e5c5 100644
--- a/graal/espresso/mx.espresso/suite.py
+++ b/graal/espresso/mx.espresso/suite.py
@@ -23,8 +23,8 @@
 suite = {
     "mxversion": "6.17.0",
     "name": "espresso",
-    "version" : "23.0.4",
-    "release" : False,
+    "version" : "23.0.5",
+    "release" : True,
     "groupId" : "org.graalvm.espresso",
     "url" : "https://www.graalvm.org/reference-manual/java-on-truffle/",
     "developer" : {
diff --git a/graal/regex/mx.regex/suite.py b/graal/regex/mx.regex/suite.py
index 5ddf3fd01b0..e3747797e89 100644
--- a/graal/regex/mx.regex/suite.py
+++ b/graal/regex/mx.regex/suite.py
@@ -43,8 +43,8 @@ suite = {
 
   "name" : "regex",
 
-  "version" : "23.0.4",
-  "release" : False,
+  "version" : "23.0.5",
+  "release" : True,
   "groupId" : "org.graalvm.regex",
   "url" : "http://www.graalvm.org/",
   "developer" : {
diff --git a/graal/sdk/mx.sdk/suite.py b/graal/sdk/mx.sdk/suite.py
index 0afe2abc071..12c5c140092 100644
--- a/graal/sdk/mx.sdk/suite.py
+++ b/graal/sdk/mx.sdk/suite.py
@@ -41,8 +41,8 @@
 suite = {
   "mxversion": "6.17.0",
   "name" : "sdk",
-  "version" : "23.0.4",
-  "release" : False,
+  "version" : "23.0.5",
+  "release" : True,
   "sourceinprojectwhitelist" : [],
   "url" : "https://github.com/oracle/graal",
   "groupId" : "org.graalvm.sdk",
diff --git a/graal/substratevm/mx.substratevm/suite.py b/graal/substratevm/mx.substratevm/suite.py
index e29ed921625..2144280f75d 100644
--- a/graal/substratevm/mx.substratevm/suite.py
+++ b/graal/substratevm/mx.substratevm/suite.py
@@ -2,8 +2,8 @@
 suite = {
     "mxversion": "6.17.0",
     "name": "substratevm",
-    "version" : "23.0.4",
-    "release" : False,
+    "version" : "23.0.5",
+    "release" : True,
     "url" : "https://github.com/oracle/graal/tree/master/substratevm",
 
     "groupId" : "org.graalvm.nativeimage",
diff --git a/graal/tools/mx.tools/suite.py b/graal/tools/mx.tools/suite.py
index 81172a2726f..e090ca0d40d 100644
--- a/graal/tools/mx.tools/suite.py
+++ b/graal/tools/mx.tools/suite.py
@@ -26,8 +26,8 @@ suite = {
     "defaultLicense" : "GPLv2-CPE",
 
     "groupId" : "org.graalvm.tools",
-    "version" : "23.0.4",
-    "release" : False,
+    "version" : "23.0.5",
+    "release" : True,
     "url" : "http://openjdk.java.net/projects/graal",
     "developer" : {
         "name" : "GraalVM Development",
diff --git a/graal/truffle/mx.truffle/suite.py b/graal/truffle/mx.truffle/suite.py
index 1f816e8e86e..fa22ce2f06b 100644
--- a/graal/truffle/mx.truffle/suite.py
+++ b/graal/truffle/mx.truffle/suite.py
@@ -41,8 +41,8 @@
 suite = {
   "mxversion": "6.17.0",
   "name" : "truffle",
-  "version" : "23.0.4",
-  "release" : False,
+  "version" : "23.0.5",
+  "release" : True,
   "groupId" : "org.graalvm.truffle",
   "sourceinprojectwhitelist" : [],
   "url" : "http://openjdk.java.net/projects/graal",
diff --git a/graal/vm/mx.vm/suite.py b/graal/vm/mx.vm/suite.py
index 8d17caef609..94e65522f54 100644
--- a/graal/vm/mx.vm/suite.py
+++ b/graal/vm/mx.vm/suite.py
@@ -1,8 +1,8 @@
 suite = {
     "name": "vm",
-    "version" : "23.0.4",
+    "version" : "23.0.5",
     "mxversion": "6.17.0",
-    "release" : False,
+    "release" : True,
     "groupId" : "org.graalvm",
 
     "url" : "http://www.graalvm.org/",
