--- /dev/null
+++ rustc-src/compiler/rustc_target/src/spec/aarch64_alpaquita_linux_gnu.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::aarch64_unknown_linux_gnu::target();
+
+    base.llvm_target = "aarch64-alpaquita-linux-gnu".into();
+    base.options.vendor = "alpaquita".into();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+    base.options.need_rpath = true;
+
+    base
+}
--- /dev/null
+++ rustc-src/compiler/rustc_target/src/spec/aarch64_alpaquita_linux_musl.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::aarch64_unknown_linux_musl::target();
+
+    base.llvm_target = "aarch64-alpaquita-linux-musl".into();
+    base.options.vendor = "alpaquita".into();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+    base.options.need_rpath = true;
+
+    base
+}
--- /dev/null
+++ rustc-src/compiler/rustc_target/src/spec/x86_64_alpaquita_linux_gnu.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::x86_64_unknown_linux_gnu::target();
+
+    base.llvm_target = "x86_64-alpaquita-linux-gnu".into();
+    base.options.vendor = "alpaquita".into();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+    base.options.need_rpath = true;
+
+    base
+}
--- /dev/null
+++ rustc-src/compiler/rustc_target/src/spec/x86_64_alpaquita_linux_musl.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::x86_64_unknown_linux_musl::target();
+
+    base.llvm_target = "x86_64-alpaquita-linux-musl".into();
+    base.options.vendor = "alpaquita".into();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+    base.options.need_rpath = true;
+
+    base
+}
--- rustc-src/compiler/rustc_target/src/spec/mod.rs.orig
+++ rustc-src/compiler/rustc_target/src/spec/mod.rs
@@ -1013,6 +1013,10 @@
 }
 
 supported_targets! {
+    ("aarch64-alpaquita-linux-gnu", aarch64_alpaquita_linux_gnu),
+    ("aarch64-alpaquita-linux-musl", aarch64_alpaquita_linux_musl),
+    ("x86_64-alpaquita-linux-gnu", x86_64_alpaquita_linux_gnu),
+    ("x86_64-alpaquita-linux-musl", x86_64_alpaquita_linux_musl),
     ("x86_64-unknown-linux-gnu", x86_64_unknown_linux_gnu),
     ("x86_64-unknown-linux-gnux32", x86_64_unknown_linux_gnux32),
     ("i686-unknown-linux-gnu", i686_unknown_linux_gnu),
