From 7099f5987956d158897bbd6d3a62533b7c2fa048 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 2 Feb 2022 17:41:55 +0300
Subject: [PATCH] handle libc tags in depends/makedepends

---
 abuild.in | 27 ++++++++++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 023dd02..7165c43 100644
--- a/abuild.in
+++ b/abuild.in
@@ -205,6 +205,28 @@ verify_pkgname() {
 	return 0
 }
 
+remove_tag() {
+	local var="$1"
+	local rm_tag="$2"
+
+	echo "$var" | sed -E "s/$rm_tag:[^ ]+//g"
+}
+
+remove_libc_tags() {
+	local rm_tag=musl
+	local t
+
+	[ "$CLIBC" = "musl" ] && rm_tag="glibc"
+
+	depends="$(remove_tag "$depends" $rm_tag)"
+	makedepends="$(remove_tag "$makedepends" $rm_tag)"
+
+	for t in musl glibc; do
+		depends="${depends/$t:/}"
+		makedepends="${makedepends/$t:/}"
+	done
+}
+
 # check if apkbuild is basically sane
 default_sanitycheck() {
 	local i= j= suggestion=
@@ -1135,6 +1157,8 @@ prepare_metafiles() {
 	mkdir -p "$controldir"
 	local builddate="$SOURCE_DATE_EPOCH"
 
+	remove_libc_tags
+
 	# Fix package size on several filesystems
 	case "$(stat -f -c "%T" .)" in
 	btrfs|ecryptfs|zfs)
@@ -1896,7 +1920,7 @@ build_abuildrepo() {
 		# make sure SOURCE_DATE_EPOCH is set
 		set_source_date
 
-		for part in sanitycheck builddeps clean fetch unpack prepare mkusers build \
+		for part in remove_libc_tags sanitycheck builddeps clean fetch unpack prepare mkusers build \
 				$_check rootpkg; do
 			runpart $part
 		done
@@ -2554,6 +2578,7 @@ rootbld() {
 	set_source_date
 
 	# networking business
+	remove_libc_tags
 	sanitycheck
 	clean
 	fetch
-- 
2.41.0

