From 3c0e7adb6f2dfa83caec8256e9c2fab00d3b2d10 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 2 Feb 2022 17:41:55 +0300
Subject: [PATCH] handle libc tags in depends/makedepends

---
 abuild.in | 27 ++++++++++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 57fc0e1..4f144bc 100644
--- a/abuild.in
+++ b/abuild.in
@@ -202,6 +202,28 @@ verify_pkgname() {
 	return 0
 }
 
+remove_tagged_items() {
+	local var="$1"
+	local rm_tag="$2"
+
+	echo "$var" | sed -E "s/$rm_tag:[^ ]+//g"
+}
+
+remove_libc_tags() {
+	local rm_tag=musl
+	local t
+
+	[ "$CLIBC" = "musl" ] && rm_tag="glibc"
+
+	depends="$(remove_tagged_items "$depends" $rm_tag)"
+	makedepends="$(remove_tagged_items "$makedepends" $rm_tag)"
+
+	for t in musl glibc; do
+		depends="${depends//$t:/}"
+		makedepends="${makedepends//$t:/}"
+	done
+}
+
 # check if apkbuild is valid
 default_validate() {
 	local i= j=
@@ -1137,6 +1159,8 @@ prepare_metafiles() {
 	mkdir -p "$controldir"
 	local builddate="$SOURCE_DATE_EPOCH"
 
+	remove_libc_tags
+
 	# Fix package size on several filesystems
 	case "$(stat -f -c "%T" .)" in
 	btrfs|ecryptfs|zfs)
@@ -1913,7 +1937,7 @@ build_abuildrepo() {
 		# make sure SOURCE_DATE_EPOCH is set
 		set_source_date
 
-		for part in validate builddeps clean fetch unpack prepare mkusers build \
+		for part in remove_libc_tags validate builddeps clean fetch unpack prepare mkusers build \
 				$_check rootpkg; do
 			runpart $part
 		done
@@ -2542,6 +2566,7 @@ rootbld() {
 	set_source_date
 
 	# networking business
+	remove_libc_tags
 	validate
 	clean
 	fetch
-- 
2.46.2

