From b30848b34b58d248894ddce4ac4560f922f90f8b Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Tue, 1 Mar 2022 16:59:59 +0300
Subject: [PATCH] new 'alternative' variable

---
 APKBUILD.5.scd | 6 ++++++
 abuild.in      | 9 +++++++++
 2 files changed, 15 insertions(+)

diff --git a/APKBUILD.5.scd b/APKBUILD.5.scd
index c3dd6a7..20c947b 100644
--- a/APKBUILD.5.scd
+++ b/APKBUILD.5.scd
@@ -168,6 +168,12 @@ file.
 	Specifies packages that the package replaces. This is typically
 	used for packages renamed by upstream.
 
+*alternative*
+	Marks a package as an alternative to packages it conflicts with
+	(they have the same *provides*). When adding an alternative package,
+	it will remove the already installed alternative package that it
+	conflicts with.
+
 *subpackages*
 	Specifies subpackages or split packages built with this
 	package. Typically, this will include _$pkgname-dev_ for development
diff --git a/abuild.in b/abuild.in
index 1052a58..2ff7ce6 100755
--- a/abuild.in
+++ b/abuild.in
@@ -256,6 +256,11 @@ default_sanitycheck() {
 		die "provider_priority must be a number"
 	fi
 
+	if [ -n "$alternative" ] \
+		&& ! echo $alternative | egrep -q '^[01]$'; then
+		die "alternative must be a number: 0 or 1"
+	fi
+
 	# check pkgname and subpkgnames
 	for i in "$pkgname" $subpackages; do
 		verify_pkgname "${i%%:*}" || die "${i%%:*} is not a valid package name"
@@ -1192,6 +1197,10 @@ prepare_metafiles() {
 		echo "provider_priority = $provider_priority" >> "$pkginfo"
 	fi
 
+	if [ -n "$alternative" ]; then
+		echo "alternative = $alternative" >> "$pkginfo"
+	fi
+
 	echo "license = $(echo $license)" >> "$pkginfo"
 	for i in $replaces; do
 		echo "replaces = $i" >> "$pkginfo"
-- 
2.36.2

