From 9681f71d0cbff6306bb452dd6e4547c5a8ea338b Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 16 Mar 2022 14:50:36 +0300
Subject: [PATCH] fix busybox path

---
 abuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 9a4bbf2..ff13085 100644
--- a/abuild.in
+++ b/abuild.in
@@ -582,7 +582,7 @@ default_unpack() {
 			*.tar.xz)
 				msg "Unpacking $s..."
 				local threads_opt
-				if [ $(readlink -f $(command -v unxz)) != "/bin/busybox" ]; then
+				if [ $(readlink -f $(command -v unxz)) != "/usr/bin/busybox" ]; then
 					threads_opt="--threads=0"
 				fi
 				unxz $threads_opt -c "$s" | tar -C "$srcdir" -x || return 1;;
-- 
2.41.0

