From 29e06b6769a71957991f577294cff721a974efca Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 16 Mar 2022 14:50:36 +0300
Subject: [PATCH] fix busybox path

---
 abuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 13cf77d..5f527af 100644
--- a/abuild.in
+++ b/abuild.in
@@ -584,7 +584,7 @@ default_unpack() {
 			*.tar.xz)
 				msg "Unpacking $s..."
 				local threads_opt
-				if [ $(readlink -f $(command -v unxz)) != "/bin/busybox" ]; then
+				if [ $(readlink -f $(command -v unxz)) != "/usr/bin/busybox" ]; then
 					threads_opt="--threads=0"
 				fi
 				unxz $threads_opt -c "$s" | tar -C "$srcdir" -x || return 1;;
-- 
2.46.2

