From d4c0baef8859e6c8fc86d44b464b0438f4eb3339 Mon Sep 17 00:00:00 2001
From: Denis Kononenko <denis.kononenko@bell-sw.com>
Date: Wed, 15 Jun 2022 10:16:32 +0000
Subject: [PATCH] Don't add /bin/sh dependency for glibc-bin.trigger

---
 abuild.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 5f527af..496c1c3 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1222,7 +1222,8 @@ prepare_metafiles() {
 	EOF
 	local deps
 	deps="$depends"
-	if [ "$pkgname" != "busybox" ] && ! depends_has busybox && ! depends_has /bin/sh; then
+	if [ "$pkgname" != "busybox" -a "$subpkgname" != "glibc-utils" ] && ! depends_has busybox \
+		&& ! depends_has /bin/sh; then
 		for i in $install $triggers; do
 			local s=${i%=*}
 			[ "$name" != "${s%.*}" ] && continue
-- 
2.46.2

