From ef8a46389964743f30e5277ba96455b12cc8433c Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Thu, 8 Sep 2022 16:28:03 +0300
Subject: [PATCH] create common vendor variables

---
 abuild.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/abuild.in b/abuild.in
index ea255ad..a36767f 100644
--- a/abuild.in
+++ b/abuild.in
@@ -28,6 +28,10 @@ fi
 : ${APK:="apk"}
 : ${ADDUSER:="abuild-adduser"}
 : ${ADDGROUP:="abuild-addgroup"}
+: ${VENDOR_NAME:="BellSoft"}
+: ${VENDOR_URL:="https://bell-sw.com/"}
+: ${VENDOR_BUG_REPORT_URL:="https://bell-sw.com/support/"}
+: ${VENDOR_JAVA_NAME:="Liberica"}
 
 apk_opt_wait="--wait 30"
 doc_threshold=$((2 * 1024 * 1024)) # 2 MiB
-- 
2.46.2

