From 6f0a054db4febf921b2840361df6b6ee28da29ff Mon Sep 17 00:00:00 2001
From: Dmitry Klochkov <dmitry.klochkov@bell-sw.com>
Date: Fri, 2 Sep 2022 19:20:01 +0300
Subject: [PATCH] abuild: warn if /bin, /sbin or /lib is found

---
 abuild.in         | 11 +++++++++
 tests/abuild_test | 60 ++++++++++++++++++++++++++++++++++++++++++++++-
 2 files changed, 70 insertions(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index b7f8784..e6280d0 100755
--- a/abuild.in
+++ b/abuild.in
@@ -832,6 +832,17 @@ postcheck() {
 		fi
 	fi
 
+	# checking for the FHS compat
+	for i in bin sbin lib; do
+		if [ -e "$dir/$i" ]; then
+			# ignore if it is a valid symlink
+			[ -h "$dir/$i" ] && [ $(readlink "$dir/$i") = "usr/$i" ] \
+				&& continue
+
+			warning "Packages must not put anything under /$i, use /usr/$i instead"
+		fi
+	done
+
 	# remove *.la files if libtool is not set
 	if ! options_has "libtool"; then
 		find "$dir" -name '*.la' -type f -delete
diff --git a/tests/abuild_test b/tests/abuild_test
index fb20246..2f78eff 100755
--- a/tests/abuild_test
+++ b/tests/abuild_test
@@ -39,7 +39,8 @@ init_tests \
 	abuild_cleanoldpkg \
 	abuild_path_with_spaces \
 	abuild_pyc_warn \
-	abuild_pyc
+	abuild_pyc \
+	abuild_invalid_fhs
 
 export ABUILD_SHAREDIR=$(atf_get_srcdir)/..
 export ABUILD_CONF=/dev/null
@@ -922,3 +923,60 @@ abuild_pyc_body() {
 	atf_check -o match:"foo=1.0-r0" -o match:"pyc" \
 		grep install_if pkg/.control.foo-pyc/.PKGINFO
 }
+
+abuild_invalid_fhs_body() {
+	init_keys
+	mkdir invalid-fhs
+	cd invalid-fhs
+
+	cat >APKBUILD.in<<-EOF
+		# Maintainer: John Doe <info@example.com>
+		pkgname="invalid-fhs"
+		pkgver="1.0"
+		pkgrel=0
+		pkgdesc="Dummy test package"
+		url="https://example.com"
+		arch="noarch"
+		license="MIT"
+		prepare() {
+			mkdir -p "\$builddir"
+		}
+		build() {
+			local i; for i in bin sbin lib; do
+				case @ftype@ in
+				link)
+					mkdir -p usr/\$i
+					ln -s usr/\$i \$i
+					;;
+				dir) mkdir \$i;;
+				esac
+			done
+		}
+		check() {
+			true
+		}
+		package() {
+			mkdir -p "\$pkgdir"
+			cp -r * "\$pkgdir"/
+		}
+	EOF
+
+	msg_fmt='Packages must not put anything under /%s, use /usr/%s instead'
+	# We need to exactly specify a keys directory for apk, otherwise
+	# command 'abuild cleanpkg' will fail.
+	export ABUILD_APK_INDEX_OPTS="--keys-dir $(readlink -f ../.abuild)"
+
+	sed 's#@ftype@#dir#' APKBUILD.in >APKBUILD
+	atf_check -s exit:0 \
+		-e match:"$(printf "$msg_fmt" bin bin)" \
+		-e match:"$(printf "$msg_fmt" sbin sbin)" \
+		-e match:"$(printf "$msg_fmt" lib lib)" \
+		abuild all
+
+	sed 's#@ftype@#link#' APKBUILD.in >APKBUILD
+	atf_check -s exit:0 \
+		-e not-match:"$(printf "$msg_fmt" bin bin)" \
+		-e not-match:"$(printf "$msg_fmt" sbin sbin)" \
+		-e not-match:"$(printf "$msg_fmt" lib lib)" \
+		abuild cleanpkg all
+}
-- 
2.39.2

