From 1603264b7c718d89d2bd0e27d8f4c94d72addbc7 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 1 Feb 2023 16:02:25 +0300
Subject: [PATCH] auto-provides: skip debuginfo files

---
 abuild.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/abuild.in b/abuild.in
index c90c264..8f8534c 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1705,6 +1705,10 @@ scan_shared_objects() {
 			soname=${soname##*/}
 		fi
 
+		suffix="${soname##*.}"
+		# skip adding debuginfo files to provides-so
+		[ "$suffix" = "debug" -o "$suffix" = "debuginfo" ] && continue
+
 		# we only want shared libs
 		case $soname in
 		*.o) continue;;
-- 
2.34.1

