From 2f3c704e006f7ad3842072ac7b23c72b0873b64b Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 1 Feb 2023 16:02:25 +0300
Subject: [PATCH] auto-provides: skip debuginfo files

---
 abuild.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/abuild.in b/abuild.in
index 02cb8dd..bf3fee2 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1719,6 +1719,10 @@ scan_shared_objects() {
 			soname=${soname##*/}
 		fi
 
+		suffix="${soname##*.}"
+		# skip adding debuginfo files to provides-so
+		[ "$suffix" = "debug" -o "$suffix" = "debuginfo" ] && continue
+
 		# we only want shared libs
 		case $soname in
 		*.o) continue;;
-- 
2.55.0

