From 83a97c0c0b909391218ea0fdf4d9ed05c6ac0449 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 1 Feb 2023 16:02:25 +0300
Subject: [PATCH] auto-provides: skip debuginfo files

---
 abuild.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/abuild.in b/abuild.in
index 994af2d..c810f1c 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1741,6 +1741,10 @@ scan_shared_objects() {
 			soname=${soname##*/}
 		fi
 
+		suffix="${soname##*.}"
+		# skip adding debuginfo files to provides-so
+		[ "$suffix" = "debug" -o "$suffix" = "debuginfo" ] && continue
+
 		# we only want shared libs
 		case $soname in
 		*.o) continue;;
-- 
2.49.0

