From b6ac1fb6cad0deacde3302d7ec446aa4270ce533 Mon Sep 17 00:00:00 2001
From: Stanislav Kholmanskikh <stanislav.kholmanskikh@bell-sw.com>
Date: Wed, 18 Jan 2023 10:43:04 +0300
Subject: [PATCH] Add distro vendor variables

---
 abuild.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/abuild.in b/abuild.in
index 9432618..15ccc0a 100644
--- a/abuild.in
+++ b/abuild.in
@@ -34,6 +34,8 @@ fi
 : ${VENDOR_NAME:="BellSoft"}
 : ${VENDOR_URL:="https://bell-sw.com/"}
 : ${VENDOR_BUG_REPORT_URL:="https://bell-sw.com/support/"}
+: ${VENDOR_DISTRO:="alpaquita"}
+: ${VENDOR_DISTRO_NAME:="Alpaquita"}
 : ${VENDOR_JAVA_NAME:="Liberica"}
 
 apk_opt_wait="--wait 30"
-- 
2.42.0

