From cc650455a04e911c9c0cb8432d9dba4ce98c73ff Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 1 Feb 2023 16:02:25 +0300
Subject: [PATCH] auto-provides: skip debuginfo files

---
 abuild.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/abuild.in b/abuild.in
index e0fe08d..c318087 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1726,6 +1726,10 @@ scan_shared_objects() {
 			soname=${soname##*/}
 		fi
 
+		suffix="${soname##*.}"
+		# skip adding debuginfo files to provides-so
+		[ "$suffix" = "debug" -o "$suffix" = "debuginfo" ] && continue
+
 		# we only want shared libs
 		case $soname in
 		*.o) continue;;
-- 
2.47.0

