From f897fbc1f9b670f05b003aaceffc6d5377aa520a Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 1 Feb 2023 16:02:25 +0300
Subject: [PATCH] auto-provides: skip debuginfo files

---
 abuild.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/abuild.in b/abuild.in
index 69d4db0..e8754e1 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1729,6 +1729,10 @@ scan_shared_objects() {
 			soname=${soname##*/}
 		fi
 
+		suffix="${soname##*.}"
+		# skip adding debuginfo files to provides-so
+		[ "$suffix" = "debug" -o "$suffix" = "debuginfo" ] && continue
+
 		# we only want shared libs
 		case $soname in
 		*.so|*.so.[0-9]*|*.c32);;
-- 
2.41.0

