From 7f4a1ef87713c61de5c2fcb76404641deff7c714 Mon Sep 17 00:00:00 2001
From: Dmitry Klochkov <dmitry.klochkov@bell-sw.com>
Date: Thu, 7 Dec 2023 19:43:25 +0300
Subject: [PATCH] abuild: add option -X for skipping running the check command
 by default

---
 abuild.in | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/abuild.in b/abuild.in
index 68f2e78..b9907af 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1967,7 +1967,7 @@ build_abuildrepo() {
 	if options_has "checkroot"; then
 		_check=check_fakeroot
 	fi
-	if ! want_check; then
+	if ! want_check || [ -n "$skip_check" ]; then
 		_check=true
 	fi
 	if ! apk_up2date || [ -n "$force" ]; then
@@ -2562,7 +2562,7 @@ rootbld_actions() {
 	if options_has "checkroot"; then
 		_check=check_fakeroot
 	fi
-	if ! want_check; then
+	if ! want_check || [ -n "$skip_check" ]; then
 		_check=true
 	fi
 	for part in symlinksrc unpack prepare build $_check check_fmv rootpkg; do
@@ -2985,6 +2985,7 @@ usage() {
 		 -r  Install missing dependencies from system repository (using \$SUDO_APK)
 		 -s  Set source package destination directory
 		 -v  Verbose: show every command as it is run (very noisy)
+		 -X  Skip running the check command by default
 
 		Commands:
 		  build       Compile and install package into \$pkgdir
@@ -3019,8 +3020,8 @@ usage() {
 }
 
 APKBUILD="${APKBUILD:-./APKBUILD}"
-unset color_opt force forceroot install_deps keep keep_build nodeps quiet verbose
-while getopts ":AcdD:fFhkKmnP:qrRs:uvV" opt; do
+unset color_opt force forceroot install_deps keep keep_build nodeps quiet verbose skip_check
+while getopts ":AcdD:fFhkKmnP:qrRs:uvVX" opt; do
 	case $opt in
 		'A') echo "$CARCH"; exit 0;;
 		'c') enable_colors
@@ -3041,6 +3042,7 @@ while getopts ":AcdD:fFhkKmnP:qrRs:uvV" opt; do
 		's') SRCDEST=$OPTARG;;
 		'v') verbose="-v"; set -x;;
 		'V') echo "$program $program_version"; exit 0;;
+		'X') skip_check="-X";;
 		'?') error "Unrecognized option: $OPTARG"; usage >&2; exit 1;;
 	esac
 done
-- 
2.46.2

