From 6a681ecab2db223c7c213507c48ff8fc20bb1c01 Mon Sep 17 00:00:00 2001
From: Dmitry Klochkov <dmitry.klochkov@bell-sw.com>
Date: Thu, 7 Dec 2023 19:43:25 +0300
Subject: [PATCH] abuild: add option -X for skipping running the check command
 by default

---
 abuild.in | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/abuild.in b/abuild.in
index 1968e83..0ea68d6 100755
--- a/abuild.in
+++ b/abuild.in
@@ -1879,7 +1879,7 @@ build_abuildrepo() {
 	if options_has "checkroot"; then
 		_check=check_fakeroot
 	fi
-	if ! want_check; then
+	if ! want_check || [ -n "$skip_check" ]; then
 		_check=true
 	fi
 	if ! apk_up2date || [ -n "$force" ]; then
@@ -2471,7 +2471,7 @@ rootbld_actions() {
 	if options_has "checkroot"; then
 		_check=check_fakeroot
 	fi
-	if ! want_check; then
+	if ! want_check || [ -n "$skip_check" ]; then
 		_check=true
 	fi
 	for part in symlinksrc unpack prepare mkusers build $_check check_fmv rootpkg; do
@@ -2866,6 +2866,7 @@ usage() {
 		 -r  Install missing dependencies from system repository (using sudo)
 		 -s  Set source package destination directory
 		 -v  Verbose: show every command as it is run (very noisy)
+		 -X  Skip running the check command by default
 
 		Commands:
 		  build       Compile and install package into \$pkgdir
@@ -2901,7 +2902,7 @@ usage() {
 
 APKBUILD="${APKBUILD:-./APKBUILD}"
 unset force
-while getopts ":AcdD:fFhkKmnP:qrRs:uvV" opt; do
+while getopts ":AcdD:fFhkKmnP:qrRs:uvVX" opt; do
 	case $opt in
 		'A') echo "$CARCH"; exit 0;;
 		'c') enable_colors
@@ -2922,6 +2923,7 @@ while getopts ":AcdD:fFhkKmnP:qrRs:uvV" opt; do
 		's') SRCDEST=$OPTARG;;
 		'v') verbose="-v"; set -x;;
 		'V') echo "$program $program_version"; exit 0;;
+		'X') skip_check="-X";;
 		'?') error "Unrecognized option: $OPTARG"; usage >&2; exit 1;;
 	esac
 done
-- 
2.42.1

