--- a/abuild.in
+++ b/abuild.in
@@ -245,7 +245,9 @@ default_sanitycheck() {
 	# digit+letter+digit passes the apk2 version check
 	case $pkgver in
 		*[0-9][a-z][0-9]*)
-			die "the digit+letter+digit version format is invalid. Use suffixes instead"
+			# was "die".  for now allow these to support
+			# versions like openjdk8 8u42
+			warning "the digit+letter+digit version format is invalid. Use suffixes instead"
 			;;
 	esac
 
--- a/tests/abuild_test
+++ b/tests/abuild_test
@@ -28,7 +28,6 @@ init_tests \
 	abuild_subpackage_arch \
 	abuild_large_doc_subpackage \
 	abuild_bigdocs \
-	abuild_pkgver_digit_letter_digit \
 	abuild_multiline_license \
 	abuild_license_spdx \
 	abuild_git_ceiling \
@@ -519,13 +518,13 @@ abuild_bigdocs_body() {
 		abuild all
 }
 
-abuild_pkgver_digit_letter_digit_body() {
-	cp -ra "$testrepo"/test-pkgname .
-	cd test-pkgname
-	TESTPKGVER=1.0p1 atf_check -s exit:1 \
-		-e match:'ERROR.*version' \
-		abuild sanitycheck
-}
+#abuild_pkgver_digit_letter_digit_body() {
+#	cp -ra "$testrepo"/test-pkgname .
+#	cd test-pkgname
+#	TESTPKGVER=1.0p1 atf_check -s exit:1 \
+#		-e match:'ERROR.*version' \
+#		abuild sanitycheck
+#}
 
 abuild_multiline_license_body() {
 	cp -ra "$testrepo"/test-licenses .
