From 5919d416c72a27a08816dfdde4a377c6052daab2 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Tue, 17 Jan 2023 16:55:02 +0300
Subject: [PATCH] increase buffer size for stdio stream functions

---
 include/stdio.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/stdio.h b/include/stdio.h
index cb858618..ca5b5275 100644
--- a/include/stdio.h
+++ b/include/stdio.h
@@ -47,7 +47,7 @@ extern "C" {
 #define _IOLBF 1
 #define _IONBF 2
 
-#define BUFSIZ 1024
+#define BUFSIZ 8192
 #define FILENAME_MAX 4096
 #define FOPEN_MAX 1000
 #define TMP_MAX 10000
-- 
2.25.1

