From beb592045ed925e9851ea2d8c488b8f38a33eb42 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Wed, 26 Jun 2024 14:11:26 +0000
Subject: [PATCH] ifunc aarch64 support

---
 arch/aarch64/reloc.h | 1 +
 include/elf.h        | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/aarch64/reloc.h b/arch/aarch64/reloc.h
index b1b68c72..9bb2a80e 100644
--- a/arch/aarch64/reloc.h
+++ b/arch/aarch64/reloc.h
@@ -19,6 +19,7 @@
 #define REL_DTPOFF      R_AARCH64_TLS_DTPREL64
 #define REL_TPOFF       R_AARCH64_TLS_TPREL64
 #define REL_TLSDESC     R_AARCH64_TLSDESC
+#define REL_IRELATIVE   R_AARCH64_IRELATIVE
 
 #define CRTJMP(pc,sp) __asm__ __volatile__( \
 	"mov sp,%1 ; br %0" : : "r"(pc), "r"(sp) : "memory" )
diff --git a/include/elf.h b/include/elf.h
index 3d5e13e4..2ce88353 100644
--- a/include/elf.h
+++ b/include/elf.h
@@ -2513,6 +2513,7 @@ enum
 #define R_AARCH64_TLS_TPREL    1030
 #define R_AARCH64_TLS_TPREL64  1030
 #define R_AARCH64_TLSDESC      1031
+#define R_AARCH64_IRELATIVE    1032
 
 
 #define R_ARM_NONE		0
-- 
2.25.1

