From b25b42b250d1ffd4303ba59f8a9c17a887bcfddc Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Thu, 29 Aug 2024 18:20:10 +0000
Subject: [PATCH] dynlink: print cpu diagnostics

---
 ldso/dynlink.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ldso/dynlink.c b/ldso/dynlink.c
index f0df3573..c4ac27dc 100644
--- a/ldso/dynlink.c
+++ b/ldso/dynlink.c
@@ -48,6 +48,7 @@ static void (*error)(const char *, ...) = error_noop;
 
 /* provided by glibc-string library */
 extern void *__get_cpu_features(void);
+extern void __print_cpu_diagnostics(void);
 
 struct debug {
 	int ver;
@@ -1882,6 +1883,9 @@ void __dls3(size_t *sp, size_t *auxv)
 				if (opt[5]=='=') replace_argv0 = opt+6;
 				else if (opt[5]) *argv = 0;
 				else if (*argv) replace_argv0 = *argv++;
+			} else if (!memcmp(opt, "list-diagnostics", 17)) {
+				__print_cpu_diagnostics();
+				_exit(0);
 			} else {
 				argv[0] = 0;
 			}
-- 
2.25.1

