From 40cef9239892c27c309e35df366650bb3f2b62b4 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Tue, 14 Dec 2021 19:55:15 +0300
Subject: [PATCH] efi/wrapper.h: make compatible with musl

---
 efi/wrapper.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/efi/wrapper.h b/efi/wrapper.h
index 0e6b38ea..c8684de1 100644
--- a/efi/wrapper.h
+++ b/efi/wrapper.h
@@ -26,6 +26,12 @@
 #define __packed	__attribute__((packed))
 #define OFFSETOF(t,m)	((size_t)&((t *)0)->m)
 
+#include <stdint.h>
+typedef uint8_t   __uint8_t;
+typedef uint16_t  __uint16_t;
+typedef uint32_t  __uint32_t;
+typedef uint64_t  __uint64_t;
+
 struct header {
 	__uint16_t msdos_signature;
 	__uint8_t _pad1[0x16];
-- 
2.25.1

