diff --git a/cloudinit/config/cc_ca_certs.py b/cloudinit/config/cc_ca_certs.py
index 302a67a4..6b35f866 100644
--- a/cloudinit/config/cc_ca_certs.py
+++ b/cloudinit/config/cc_ca_certs.py
@@ -46,7 +46,7 @@ can be removed from the system with the configuration option
     ca-certificates package is installed but not if the
     ca-certificates-bundle package is installed.
 """
-distros = ["alpine", "debian", "ubuntu", "rhel"]
+distros = ["alpine", "alpaquita", "debian", "ubuntu", "rhel"]
 
 meta: MetaSchema = {
     "id": "cc_ca_certs",
diff --git a/cloudinit/config/cc_ntp.py b/cloudinit/config/cc_ntp.py
index 8ecc4eb8..15a0bdbe 100644
--- a/cloudinit/config/cc_ntp.py
+++ b/cloudinit/config/cc_ntp.py
@@ -25,6 +25,7 @@ NTP_CONF = "/etc/ntp.conf"
 NR_POOL_SERVERS = 4
 distros = [
     "almalinux",
+    "alpaquita",
     "alpine",
     "centos",
     "cloudlinux",
@@ -92,6 +93,17 @@ NTP_CLIENT_CONFIG = {
 
 # This is Distro-specific configuration overrides of the base config
 DISTRO_CLIENT_CONFIG = {
+    "alpaquita": {
+        "chrony": {
+            "confpath": "/etc/chrony/chrony.conf",
+            "service_name": "chronyd",
+        },
+        "ntp": {
+            "confpath": "/etc/ntp.conf",
+            "packages": [],
+            "service_name": "ntpd",
+        }
+    },
     "alpine": {
         "chrony": {
             "confpath": "/etc/chrony/chrony.conf",
@@ -394,7 +406,7 @@ def generate_server_names(distro):
         # For legal reasons x.pool.sles.ntp.org does not exist,
         # use the opensuse pool
         pool_distro = "opensuse"
-    elif distro == "alpine" or distro == "eurolinux":
+    elif distro in ["alpaquita", "alpine", "eurolinux"]:
         # Alpine-specific pool (i.e. x.alpine.pool.ntp.org) does not exist
         # so use general x.pool.ntp.org instead. The same applies to EuroLinux
         pool_distro = ""
@@ -444,7 +456,7 @@ def write_ntp_config_template(
         return
     if (
         len(servers) == 0
-        and distro_name == "alpine"
+        and distro_name in ["alpaquita", "alpine"]
         and service_name == "ntpd"
     ):
         # Alpine's Busybox ntpd only understands "servers" configuration
diff --git a/cloudinit/config/cc_resolv_conf.py b/cloudinit/config/cc_resolv_conf.py
index 8dbed71e..97b1deb2 100644
--- a/cloudinit/config/cc_resolv_conf.py
+++ b/cloudinit/config/cc_resolv_conf.py
@@ -56,6 +56,7 @@ meta: MetaSchema = {
     "title": "Configure resolv.conf",
     "description": MODULE_DESCRIPTION,
     "distros": [
+        "alpaquita",
         "alpine",
         "fedora",
         "mariner",
diff --git a/cloudinit/distros/__init__.py b/cloudinit/distros/__init__.py
index 350b7e28..f606244d 100644
--- a/cloudinit/distros/__init__.py
+++ b/cloudinit/distros/__init__.py
@@ -41,7 +41,7 @@ from cloudinit.net.renderer import Renderer
 ALL_DISTROS = "all"
 
 OSFAMILIES = {
-    "alpine": ["alpine"],
+    "alpine": ["alpine", "alpaquita"],
     "arch": ["arch"],
     "debian": ["debian", "ubuntu"],
     "freebsd": ["freebsd", "dragonfly"],
diff --git a/cloudinit/util.py b/cloudinit/util.py
index 078ce1c2..cbc9664c 100644
--- a/cloudinit/util.py
+++ b/cloudinit/util.py
@@ -629,6 +629,7 @@ def _get_variant(info):
         linux_dist = info["dist"][0].lower()
         if linux_dist in (
             "almalinux",
+            "alpaquita",
             "alpine",
             "arch",
             "centos",
diff --git a/config/cloud.cfg.tmpl b/config/cloud.cfg.tmpl
index 13cdd695..dd978bf7 100644
--- a/config/cloud.cfg.tmpl
+++ b/config/cloud.cfg.tmpl
@@ -34,7 +34,7 @@ disable_root: false
 disable_root: true
 {% endif %}
 
-{% if variant in ["almalinux", "alpine", "amazon", "cloudlinux", "eurolinux",
+{% if variant in ["almalinux", "alpaquita", "alpine", "amazon", "cloudlinux", "eurolinux",
                   "fedora", "miraclelinux", "openEuler", "openmandriva", "rocky", "virtuozzo"] or is_rhel %}
 {% if is_rhel %}
 mount_default_fields: [~, ~, 'auto', 'defaults,nofail,x-systemd.requires=cloud-init.service,_netdev', '0', '2']
@@ -99,7 +99,7 @@ cloud_init_modules:
  - set_hostname
  - update_hostname
  - update_etc_hosts
-{% if variant in ["alpine", "photon"] %}
+{% if variant in ["alpine", "alpaquita", "photon"] %}
  - resolv_conf
 {% endif %}
 {% if not variant.endswith("bsd") %}
@@ -123,7 +123,7 @@ cloud_config_modules:
  - ubuntu_autoinstall
 {% endif %}
 {% if variant not in ["photon"] %}
-{% if variant not in ["alpine"] %}
+{% if variant not in ["alpine", "alpaquita"] %}
  - ssh-import-id
 {% endif %}
 {% if variant in ["arch", "debian", "ubuntu", "suse"] %}
@@ -152,7 +152,7 @@ cloud_config_modules:
 {% if variant in ["suse"] %}
  - zypper-add-repo
 {% endif %}
-{% if variant in ["alpine"] %}
+{% if variant in ["alpine", "alpaquita"] %}
  - apk-configure
 {% endif %}
  - ntp
@@ -180,7 +180,7 @@ cloud_final_modules:
  - ansible
  - mcollective
  - salt-minion
-{% if variant not in ["alpine"] %}
+{% if variant not in ["alpine", "alpaquita"] %}
  - reset_rmc
  - refresh_rmc_and_interface
 {% endif %}
@@ -201,7 +201,7 @@ cloud_final_modules:
 # (not accessible to handlers/transforms)
 system_info:
    # This will affect which distro class gets used
-{% if variant in ["almalinux", "alpine", "amazon", "arch", "cloudlinux", "debian",
+{% if variant in ["almalinux", "alpaquita", "alpine", "amazon", "arch", "cloudlinux", "debian",
                   "eurolinux", "fedora", "freebsd", "gentoo", "netbsd", "mariner", "miraclelinux", "openbsd", "openEuler",
                   "openmandriva", "photon", "rocky", "suse", "ubuntu", "virtuozzo"] or is_rhel %}
    distro: {{ variant }}
@@ -256,7 +256,7 @@ system_info:
          primary: http://ports.ubuntu.com/ubuntu-ports
          security: http://ports.ubuntu.com/ubuntu-ports
    ssh_svcname: ssh
-{% elif variant in ["almalinux", "alpine", "amazon", "arch", "cloudlinux", "eurolinux",
+{% elif variant in ["almalinux", "alpaquita", "alpine", "amazon", "arch", "cloudlinux", "eurolinux",
                     "fedora", "gentoo", "miraclelinux", "openEuler", "openmandriva", "rocky", "suse", "virtuozzo"] or is_rhel %}
    # Default user name + that default users groups (if added/used)
    default_user:
@@ -286,7 +286,7 @@ system_info:
      groups: [users, wheel]
      primary_group: users
      no_user_group: true
-{% elif variant == "alpine" %}
+{% elif variant in ["alpine", "alpaquita"] %}
      groups: [adm, sudo, wheel]
 {% elif variant == "arch" %}
      groups: [wheel, users]
@@ -298,7 +298,7 @@ system_info:
      groups: [wheel, adm, systemd-journal]
 {% endif %}
      sudo: ["ALL=(ALL) NOPASSWD:ALL"]
-{% if variant == "alpine" %}
+{% if variant in ["alpine", "alpaquita"] %}
      shell: /bin/ash
 {% else %}
      shell: /bin/bash
diff --git a/tests/unittests/config/test_cc_ntp.py b/tests/unittests/config/test_cc_ntp.py
index 365945f8..2e14c972 100644
--- a/tests/unittests/config/test_cc_ntp.py
+++ b/tests/unittests/config/test_cc_ntp.py
@@ -250,7 +250,7 @@ class TestNtp(FilesystemMockingTestCase):
 
     def _get_expected_pools(self, pools, distro, client):
         if client in ["ntp", "chrony"]:
-            if client == "ntp" and distro == "alpine":
+            if client == "ntp" and distro in ["alpine", "alpaquita"]:
                 # NTP for Alpine Linux is Busybox's ntp which does not
                 # support 'pool' lines in its configuration file.
                 expected_pools = []
@@ -265,7 +265,7 @@ class TestNtp(FilesystemMockingTestCase):
 
     def _get_expected_servers(self, servers, distro, client):
         if client in ["ntp", "chrony"]:
-            if client == "ntp" and distro == "alpine":
+            if client == "ntp" and distro in ["alpine", "alpaquita"]:
                 # NTP for Alpine Linux is Busybox's ntp which only supports
                 # 'server' lines without iburst option.
                 expected_servers = [
@@ -381,7 +381,7 @@ class TestNtp(FilesystemMockingTestCase):
                 confpath = ntpconfig["confpath"]
                 m_select.return_value = ntpconfig
                 cc_ntp.handle("cc_ntp", valid_empty_config, mycloud, None, [])
-                if distro == "alpine":
+                if distro in ["alpine", "alpaquita"]:
                     # _mock_ntp_client_config call above did not specify a
                     # client value and so it defaults to "ntp" which on
                     # Alpine Linux only supports servers and not pools.
@@ -461,7 +461,7 @@ class TestNtp(FilesystemMockingTestCase):
             if distro == "cos":
                 return
 
-            if distro == "alpine":
+            if distro in ["alpine", "alpaquita"]:
                 uses_systemd = False
                 expected_service_call = [
                     "rc-service",
diff --git a/tests/unittests/test_cli.py b/tests/unittests/test_cli.py
index dd85a1c7..c1524159 100644
--- a/tests/unittests/test_cli.py
+++ b/tests/unittests/test_cli.py
@@ -246,7 +246,7 @@ class TestCLI:
                 ["all"],
                 [
                     "**Supported distros:** all",
-                    "**Supported distros:** almalinux, alpine, centos, "
+                    "**Supported distros:** almalinux, alpaquita, alpine, centos, "
                     "cloudlinux, cos, debian, eurolinux, fedora, freebsd, "
                     "mariner, miraclelinux, "
                     "openbsd, openEuler, openmandriva, "
diff --git a/tests/unittests/test_util.py b/tests/unittests/test_util.py
index 0c2735ae..e244046e 100644
--- a/tests/unittests/test_util.py
+++ b/tests/unittests/test_util.py
@@ -1235,6 +1235,7 @@ class TestGetVariant:
         "info, expected_variant",
         [
             ({"system": "Linux", "dist": ("almalinux",)}, "almalinux"),
+            ({"system": "linux", "dist": ("alpaquita",)}, "alpaquita"),
             ({"system": "linux", "dist": ("alpine",)}, "alpine"),
             ({"system": "linux", "dist": ("arch",)}, "arch"),
             ({"system": "linux", "dist": ("centos",)}, "centos"),
diff --git a/tools/render-cloudcfg b/tools/render-cloudcfg
index c04daead..571c71f1 100755
--- a/tools/render-cloudcfg
+++ b/tools/render-cloudcfg
@@ -11,6 +11,7 @@ def main():
 
     VARIANTS = [
         "almalinux",
+        "alpaquita",
         "alpine",
         "amazon",
         "arch",
