--- a/tests/stat-test
+++ b/tests/stat-test
@@ -28,7 +28,7 @@
 if ! [[ $OPTS == *"redhat-hardened-cc1"* ]];
 then
     echo " $TEST_NAME: using built-in option selection instead"
-    OPTS="-O2 -fexceptions -g -grecord-gcc-switches -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fPIE -fstack-protector-strong -fasynchronous-unwind-tables -fstack-clash-protection -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -pie"
+    OPTS="-O2 -fexceptions -g -grecord-gcc-switches -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fPIE -fstack-protector-strong -fasynchronous-unwind-tables -fstack-clash-protection -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -pie -D_LARGEFILE64_SOURCE"
 fi
 
 # Now add in our newly built plugin.
