From b2436ee19e43fc8e934278a5bd81ed5c1017018a Mon Sep 17 00:00:00 2001
From: Denis Kononenko <denis.kononenko@bell-sw.com>
Date: Wed, 13 Oct 2021 23:48:32 +0000
Subject: [PATCH 3/3] mx: fix for local ninja

---
 mx.mx/suite.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mx.mx/suite.py b/mx.mx/suite.py
index 48444a5..04436f4 100644
--- a/mx/mx.mx/suite.py
+++ b/mx/mx.mx/suite.py
@@ -606,6 +606,9 @@ suite = {
           "amd64" : {
             "urls" : ["https://lafo.ssw.jku.at/pub/graal-external-deps/ninja-{version}-linux-amd64-musl.zip"],
             "sha1" : "0cc6d5cff72e63444b7abb3fc0562f6e70089147"
+          },
+          "<others>" : {
+            "optional" : True
           }
         },
         "darwin" : {
-- 
2.30.2

