diff --git a/substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/AbstractMemoryPoolMXBean.java b/substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/AbstractMemoryPoolMXBean.java
index f2a23be29ba..d155352a01b 100644
--- a/graal/substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/AbstractMemoryPoolMXBean.java
+++ b/graal/substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/AbstractMemoryPoolMXBean.java
@@ -45,12 +45,14 @@ import sun.management.Util;
 
 public abstract class AbstractMemoryPoolMXBean extends AbstractMXBean implements MemoryPoolMXBean {
 
+    protected static final UnsignedWord UNDEFINED = WordFactory.signed(UNDEFINED_MEMORY_USAGE);
+    private static final UnsignedWord UNINITIALIZED = WordFactory.zero();
+
     private final String name;
     private final String[] managerNames;
     protected final UninterruptibleUtils.AtomicUnsigned peakUsage = new UninterruptibleUtils.AtomicUnsigned();
 
-    private static final UnsignedWord UNDEFINED = WordFactory.zero();
-    protected UnsignedWord initialValue = UNDEFINED;
+    protected UnsignedWord initialValue = UNINITIALIZED;
 
     @Platforms(Platform.HOSTED_ONLY.class)
     protected AbstractMemoryPoolMXBean(String name, String... managerNames) {
@@ -59,7 +61,7 @@ public abstract class AbstractMemoryPoolMXBean extends AbstractMXBean implements
     }
 
     UnsignedWord getInitialValue() {
-        if (initialValue.equal(UNDEFINED)) {
+        if (initialValue.equal(UNINITIALIZED)) {
             initialValue = computeInitialValue();
         }
         return initialValue;
diff --git a/substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/GenScavengeMemoryPoolMXBeans.java b/substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/GenScavengeMemoryPoolMXBeans.java
index 441b4e6d6b3..53216cb9166 100644
--- a/graal/substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/GenScavengeMemoryPoolMXBeans.java
+++ b/graal/substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/GenScavengeMemoryPoolMXBeans.java
@@ -81,7 +81,7 @@ public class GenScavengeMemoryPoolMXBeans {
 
         @Override
         UnsignedWord getMaximumValue() {
-            return GCImpl.getPolicy().getMaximumEdenSize();
+            return UNDEFINED;
         }
 
         @Override
@@ -124,7 +124,7 @@ public class GenScavengeMemoryPoolMXBeans {
 
         @Override
         UnsignedWord getMaximumValue() {
-            return GCImpl.getPolicy().getMaximumSurvivorSize();
+            return UNDEFINED;
         }
 
         @Override
@@ -162,7 +162,7 @@ public class GenScavengeMemoryPoolMXBeans {
 
         @Override
         UnsignedWord getMaximumValue() {
-            return GCImpl.getPolicy().getMaximumOldSize();
+            return UNDEFINED;
         }
 
         @Override
@@ -200,7 +200,7 @@ public class GenScavengeMemoryPoolMXBeans {
 
         @Override
         UnsignedWord getMaximumValue() {
-            return GCImpl.getPolicy().getMaximumHeapSize();
+            return UNDEFINED;
         }
 
         @Override
