diff --git a/vm/mx.vm/ce b/vm/mx.vm/ce
index 17ac29f6972..07b736dfd42 100644
--- a/graal/vm/mx.vm/ce
+++ b/graal/vm/mx.vm/ce
@@ -1,5 +1,5 @@
-DYNAMIC_IMPORTS=/sdk,/truffle,/compiler,/substratevm
-COMPONENTS=cmp,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm
+DYNAMIC_IMPORTS=/sdk,/truffle,/compiler,/substratevm
+COMPONENTS=cmp,gu,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm,truffle-json
 NATIVE_IMAGES=lib:jvmcicompiler,lib:native-image-agent,lib:native-image-diagnostics-agent,native-image
 NON_REBUILDABLE_IMAGES=lib:jvmcicompiler
 DISABLE_INSTALLABLES=False
diff --git a/vm/mx.vm/ce-aarch64 b/vm/mx.vm/ce-aarch64
index 17ac29f6972..4fe19734868 100644
--- a/graal/vm/mx.vm/ce-aarch64
+++ b/graal/vm/mx.vm/ce-aarch64
@@ -1,5 +1,5 @@
 DYNAMIC_IMPORTS=/sdk,/truffle,/compiler,/substratevm
-COMPONENTS=cmp,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm
+COMPONENTS=cmp,gu,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm,truffle-json
 NATIVE_IMAGES=lib:jvmcicompiler,lib:native-image-agent,lib:native-image-diagnostics-agent,native-image
 NON_REBUILDABLE_IMAGES=lib:jvmcicompiler
 DISABLE_INSTALLABLES=False
diff --git a/vm/mx.vm/ce-darwin b/vm/mx.vm/ce-darwin
index 17ac29f6972..4fe19734868 100644
--- a/graal/vm/mx.vm/ce-darwin
+++ b/graal/vm/mx.vm/ce-darwin
@@ -1,5 +1,5 @@
 DYNAMIC_IMPORTS=/sdk,/truffle,/compiler,/substratevm
-COMPONENTS=cmp,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm
+COMPONENTS=cmp,gu,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm,truffle-json
 NATIVE_IMAGES=lib:jvmcicompiler,lib:native-image-agent,lib:native-image-diagnostics-agent,native-image
 NON_REBUILDABLE_IMAGES=lib:jvmcicompiler
 DISABLE_INSTALLABLES=False
diff --git a/vm/mx.vm/ce-darwin-aarch64 b/vm/mx.vm/ce-darwin-aarch64
index 17ac29f6972..4fe19734868 100644
--- a/graal/vm/mx.vm/ce-darwin-aarch64
+++ b/graal/vm/mx.vm/ce-darwin-aarch64
@@ -1,5 +1,5 @@
 DYNAMIC_IMPORTS=/sdk,/truffle,/compiler,/substratevm
-COMPONENTS=cmp,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm
+COMPONENTS=cmp,gu,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm,truffle-json
 NATIVE_IMAGES=lib:jvmcicompiler,lib:native-image-agent,lib:native-image-diagnostics-agent,native-image
 NON_REBUILDABLE_IMAGES=lib:jvmcicompiler
 DISABLE_INSTALLABLES=False
diff --git a/vm/mx.vm/ce-win b/vm/mx.vm/ce-win
index 17ac29f6972..4fe19734868 100644
--- a/graal/vm/mx.vm/ce-win
+++ b/graal/vm/mx.vm/ce-win
@@ -1,5 +1,5 @@
 DYNAMIC_IMPORTS=/sdk,/truffle,/compiler,/substratevm
-COMPONENTS=cmp,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm
+COMPONENTS=cmp,gu,lg,ni,nic,nil,sdkni,svm,svmt,svmsl,tflc,tflsm,truffle-json
 NATIVE_IMAGES=lib:jvmcicompiler,lib:native-image-agent,lib:native-image-diagnostics-agent,native-image
 NON_REBUILDABLE_IMAGES=lib:jvmcicompiler
 DISABLE_INSTALLABLES=False
diff --git a/vm/mx.vm/gu.cmd b/vm/mx.vm/gu.cmd
index 7b50da46931..0d301848620 100644
--- a/graal/vm/mx.vm/gu.cmd
+++ b/graal/vm/mx.vm/gu.cmd
@@ -32,7 +32,7 @@ set location=%~dp0
 set root_dir=%location%..\..\..
 set relcp=%location%..\*.jar
 
-set realcp=%root_dir%\lib\graalvm\launcher-common.jar
+set realcp="%root_dir%\lib\graalvm\launcher-common.jar;%root_dir%\lib\graalvm\jline3.jar"
 for %%i in ("%relcp%") do (
   set realcp=!realcp!;%%i
 )
