diff --git a/graal/compiler/mx.compiler/suite.py b/graal/compiler/mx.compiler/suite.py
index 90fa7e54520..cb0b24691f9 100644
--- a/graal/compiler/mx.compiler/suite.py
+++ b/graal/compiler/mx.compiler/suite.py
@@ -4,7 +4,7 @@ suite = {
   "sourceinprojectwhitelist" : [],
 
   "groupId" : "org.graalvm.compiler",
-  "version" : "23.1.3",
+  "version" : "23.1.4",
   "release" : True,
   "url" : "http://www.graalvm.org/",
   "developer" : {
diff --git a/graal/espresso/mx.espresso/suite.py b/graal/espresso/mx.espresso/suite.py
index 4a9d87814a0..64c2680e248 100644
--- a/graal/espresso/mx.espresso/suite.py
+++ b/graal/espresso/mx.espresso/suite.py
@@ -23,7 +23,7 @@
 suite = {
     "mxversion": "6.44.0",
     "name": "espresso",
-    "version" : "23.1.3",
+    "version" : "23.1.4",
     "release" : True,
     "groupId" : "org.graalvm.espresso",
     "url" : "https://www.graalvm.org/reference-manual/java-on-truffle/",
diff --git a/graal/regex/mx.regex/suite.py b/graal/regex/mx.regex/suite.py
index df4e8fbc113..94b933a5e31 100644
--- a/graal/regex/mx.regex/suite.py
+++ b/graal/regex/mx.regex/suite.py
@@ -43,7 +43,7 @@ suite = {
 
   "name" : "regex",
 
-  "version" : "23.1.3",
+  "version" : "23.1.4",
   "release" : True,
   "groupId" : "org.graalvm.regex",
   "url" : "http://www.graalvm.org/",
diff --git a/graal/sdk/mx.sdk/suite.py b/graal/sdk/mx.sdk/suite.py
index 995839bfa0f..4cff1f99585 100644
--- a/graal/sdk/mx.sdk/suite.py
+++ b/graal/sdk/mx.sdk/suite.py
@@ -41,7 +41,7 @@
 suite = {
   "mxversion": "6.39.0",
   "name" : "sdk",
-  "version" : "23.1.3",
+  "version" : "23.1.4",
   "release" : True,
   "sourceinprojectwhitelist" : [],
   "url" : "https://github.com/oracle/graal",
diff --git a/graal/substratevm/mx.substratevm/suite.py b/graal/substratevm/mx.substratevm/suite.py
index dff8038e303..65068e28727 100644
--- a/graal/substratevm/mx.substratevm/suite.py
+++ b/graal/substratevm/mx.substratevm/suite.py
@@ -2,7 +2,7 @@
 suite = {
     "mxversion": "6.27.1",
     "name": "substratevm",
-    "version" : "23.1.3",
+    "version" : "23.1.4",
     "release" : True,
     "url" : "https://github.com/oracle/graal/tree/master/substratevm",
 
diff --git a/graal/tools/mx.tools/suite.py b/graal/tools/mx.tools/suite.py
index 6a7f4b91feb..767c9f73271 100644
--- a/graal/tools/mx.tools/suite.py
+++ b/graal/tools/mx.tools/suite.py
@@ -26,7 +26,7 @@ suite = {
     "defaultLicense" : "GPLv2-CPE",
 
     "groupId" : "org.graalvm.tools",
-    "version" : "23.1.3",
+    "version" : "23.1.4",
     "release" : True,
     "url" : "http://openjdk.java.net/projects/graal",
     "developer" : {
diff --git a/graal/truffle/mx.truffle/suite.py b/graal/truffle/mx.truffle/suite.py
index 8808019f129..e6157de6442 100644
--- a/graal/truffle/mx.truffle/suite.py
+++ b/graal/truffle/mx.truffle/suite.py
@@ -41,7 +41,7 @@
 suite = {
   "mxversion": "6.39.0",
   "name" : "truffle",
-  "version" : "23.1.3",
+  "version" : "23.1.4",
   "release" : True,
   "groupId" : "org.graalvm.truffle",
   "sourceinprojectwhitelist" : [],
diff --git a/graal/vm/mx.vm/suite.py b/graal/vm/mx.vm/suite.py
index 1853c8663ba..3ce04f2337a 100644
--- a/graal/vm/mx.vm/suite.py
+++ b/graal/vm/mx.vm/suite.py
@@ -1,6 +1,6 @@
 suite = {
     "name": "vm",
-    "version" : "23.1.3",
+    "version" : "23.1.4",
     "mxversion": "6.41.0",
     "release" : True,
     "groupId" : "org.graalvm",
diff --git a/graal/wasm/mx.wasm/suite.py b/graal/wasm/mx.wasm/suite.py
index f9e89c44b46..ab4e9492ff0 100644
--- a/graal/wasm/mx.wasm/suite.py
+++ b/graal/wasm/mx.wasm/suite.py
@@ -42,7 +42,7 @@ suite = {
   "mxversion": "6.41.0",
   "name" : "wasm",
   "groupId" : "org.graalvm.wasm",
-  "version" : "23.1.3",
+  "version" : "23.1.4",
   "versionConflictResolution" : "latest",
   "url" : "http://graalvm.org/",
   "developer" : {
