diff --git a/sdk/mx.sdk/mx_sdk_vm_impl.py b/sdk/mx.sdk/mx_sdk_vm_impl.py
index b86b3f13f1e..09704587085 100644
--- a/graal/sdk/mx.sdk/mx_sdk_vm_impl.py
+++ b/graal/sdk/mx.sdk/mx_sdk_vm_impl.py
@@ -2470,7 +2470,7 @@ def _format_properties(data):
 
 
 def _get_component_stability(component):
-    if _src_jdk_version not in (17, 20):
+    if _src_jdk_version not in (17, 21):
         return "experimental"
     return component.stability
 
