diff --git a/mx.mx/suite.py b/mx.mx/suite.py
index a91b1624..09ba275f 100644
--- a/mx/mx.mx/suite.py
+++ b/mx/mx.mx/suite.py
@@ -600,6 +600,9 @@ suite = {
           }
         },
         "linux-musl" : {
+          "<others>" : {
+            "optional" : True
+          },
           # Steps to build:
           # (Built in an Alpine docker container, Alpine version 3.13.0)
           # apk add python2 g++ re2c git
