diff --git a/testing/cffi0/test_ownlib.py b/testing/cffi0/test_ownlib.py
index ffad879..f6392f0 100644
--- a/testing/cffi0/test_ownlib.py
+++ b/testing/cffi0/test_ownlib.py
@@ -386,6 +386,7 @@ class TestOwnLib(object):
         assert s.bottom == 44
 
     def test_dlopen_handle(self):
+        py.test.skip("musl provides no libdl.so*, only a stub libdl.a")
         if self.module is None:
             py.test.skip("fix the auto-generation of the tiny test lib")
         if sys.platform == 'win32':
diff --git a/testing/cffi1/test_re_python.py b/testing/cffi1/test_re_python.py
index 2ae0dd1..55a0555 100644
--- a/testing/cffi1/test_re_python.py
+++ b/testing/cffi1/test_re_python.py
@@ -267,6 +267,7 @@ def test_selfref():
     ffi.new("selfref_ptr_t")
 
 def test_dlopen_handle():
+    py.test.skip("musl provides no libdl.so*, only a stub libdl.a")
     import _cffi_backend
     from re_python_pysrc import ffi
     if sys.platform == 'win32':
