--- a/Lib/test/test_urllib2net.py.orig
+++ b/Lib/test/test_urllib2net.py
@@ -140,9 +140,9 @@
     def test_ftp(self):
         # Testing the same URL twice exercises the caching in CacheFTPHandler
         urls = [
-            'ftp://www.pythontest.net/README',
-            'ftp://www.pythontest.net/README',
-            ('ftp://www.pythontest.net/non-existent-file',
+            'ftp://ftp.pythontest.internal/README',
+            'ftp://ftp.pythontest.internal/README',
+            ('ftp://ftp.pythontest.internal/non-existent-file',
              None, urllib.error.URLError),
             ]
         self._test_urls(urls, self._extra_handlers())
@@ -372,7 +372,7 @@
             self.addCleanup(u.close)
             self.assertEqual(u.fp.raw._sock.gettimeout(), 120)

-    FTP_HOST = 'ftp://www.pythontest.net/'
+    FTP_HOST = 'ftp://ftp.pythontest.internal/'

     @support.requires_resource('walltime')
     def test_ftp_basic(self):
